---
title: "RNAseq Differential Expression"
output:
  pdf_document: default
  html_notebook: default
---
```{r}
BiocManager::install("Mus.musculus")
BiocManager::install("org.Hs.eg.db")
BiocManager::install("clusterProfiler")
BiocManager::install("scRNAseq")
BiocManager::install("SingleCellExperiment")
BiocManager::install("fgsea")
devtools::install_github('xuranw/MuSiC')
remotes::install_github("renozao/xbioc")

# install the MuSiC2 package
if (!"MuSiC2" %in% rownames(installed.packages())) {
  devtools::install_github('Jiaxin-Fan/MuSiC2')
}
```


The majority of Library packages are listed here. If you want to include more just install and load at the end of the list. If code isn't recognizing the function, it is likely that R didn't load the package that the function comes from.
#load libraries every time
```{r, results=FALSE}
library(ggplot2)
library(devtools)
library(rgl)
library(tidyverse)
library(tidyr)
library(RColorBrewer)
library(expss)
library(dplyr)
library(BiocManager)
library(RMariaDB)
library(GenomicAlignments)
library(BiocParallel)
library(GenomicFeatures)
library(DESeq2)
library(Mus.musculus)
library(Rsamtools)
library(pheatmap)
library(topGO)
library(gridExtra)
library(AnnotationDbi)
library(org.Mm.eg.db)
library(EnhancedVolcano)
library(TxDb.Mmusculus.UCSC.mm10.knownGene)
library(enrichplot)
library(clusterProfiler)
library(pathview)
library(readr)
#library(MuSiC2)
library(ggpubr)
library(ggsignif)
library(rstatix)
library(scRNAseq)
library(VennDiagram)
library(fgsea)
library(stringr)
library(tibble)
library(viridis)
library(DOSE)
library(babelgene)
library(patchwork)
library(org.Hs.eg.db)
```

Set your working directory and pull in your reference genome to compare your samples to. We will read the gene model from an Ensembl GTF file. Be sure your gene model reflects the same model used in your samples. 

***Define the Gene Model***
```{r}
setwd("P:/OEH/Faculty/hlehmler/research/Lehmler Group-Amanda Bullert/MARBLE/rawdata/") #setting directory
TxDb<-makeTxDbFromUCSC(genome = "mm10",
                        tablename ="knownGene", 
                        goldenPath.url = getOption("UCSC.goldenPath.url"))
```

For additional notes reference "https://www.bioconductor.org/help/course-materials/2016/CSAMA/lab-3-rnaseq/rnaseq_gene_CSAMA2016.html"

The following line produces a GRangesList of all the exons grouped by gene (Lawrence et al. 2013). Each element of the list is a GRanges object of the exons for a gene
```{r}
ebg <- exonsBy(TxDb, 
               by="gene")
ebg
g_ids<-names(ebg)
```

Now that our gene model reference is ready we can load in samples and analyze them accordingly. I found separating by tissue type was easiest. But it can also be helpful to load all samples together. 

***Liver Samples***
```{r}
Livinfo<- read.csv(file = "Liv_sample_info.csv", #reading in the sample information
                   header = T, 
                   sep = ",")

liv_filenames <- file.path("P:/OEH/Faculty/hlehmler/research/Lehmler Group-Amanda Bullert/MARBLE/rawdata/BAM_files/", 
                           Livinfo$Sample_ID)
file.exists(liv_filenames) #a sanity check to make sure you have files named
```

Next specify the details about how the BAM files should be treated in R, e.g., only process 2 million reads at a time. This can be modified to your computing limitations.
```{r}
liv_bamfiles <- BamFileList(liv_filenames, 
                            yieldSize=2000000)
seqinfo(liv_bamfiles)
```

#Counting genes
##liver
```{r}

liv_se <- summarizeOverlaps(features=ebg, #define the gene reference
                            reads=liv_bamfiles, #samples to be read
                            mode="Union", 
                            singleEnd=FALSE, #False indicates samples are paired-end
                            ignore.strand=FALSE, #not a strand specific experiment
                            BPPARAM= SerialParam(progressbar = TRUE)) #progress bar shown
liv_se
head(assay(liv_se)) #access the counts
str(metadata(rowRanges(liv_se))) #just to look a the structure of data

```

##striatum
```{r}
Sinfo<- read.csv(file = "striatum_sample_info.csv", header = T, sep = ",")
s_filenames <- file.path("P:/OEH/Faculty/hlehmler/research/Lehmler Group-Amanda Bullert/MARBLE/rawdata/BAM_files/", Sinfo$Sample_ID)
file.exists(s_filenames)
```

```{r}
s_bamfiles <- BamFileList(s_filenames, 
                            yieldSize=2000000)
seqinfo(s_bamfiles)
```

```{r}
s_se <- summarizeOverlaps(features=ebg, 
                            reads=s_bamfiles,
                            mode="Union",
                            singleEnd=FALSE,
                            ignore.strand=FALSE,
                            BPPARAM=SerialParam(progressbar = TRUE))
s_se
head(assay(s_se))
str(metadata(rowRanges(s_se)))
```

##prefrontal cortex
```{r}
pinfo<- read.csv(file = "piriform_sample_info.csv", 
                 header = T, 
                 sep = ",")
p_filenames <- file.path("P:/OEH/Faculty/hlehmler/research/Lehmler Group-Amanda Bullert/MARBLE/rawdata/BAM_files/", 
                         pinfo$Sample_ID)
file.exists(p_filenames)
```

```{r}
p_bamfiles <- BamFileList(p_filenames, 
                            yieldSize=2000000)
seqinfo(p_bamfiles)
```

```{r}
p_se <- summarizeOverlaps(features=ebg, 
                            reads=p_bamfiles,
                            mode="Union",
                            singleEnd=FALSE,
                            ignore.strand=FALSE,
                            BPPARAM=SerialParam(progressbar = TRUE))
p_se
head(assay(p_se))
str(metadata(rowRanges(p_se)))
```

#***proceed from here*** 
(no need to recount summarized experiments)
```{r}
Livinfo$samples<- gsub(".bam", "", Livinfo$ï..Sample_ID)
rownames(Livinfo)<-Livinfo$samples

colData(liv_se) #metadata about the samples
colData(liv_se)<-DataFrame(Livinfo) #take the sample info and assign it as the metadata
liv_se$Treatment<- as.factor(liv_se$Treatment) #organizing structure of groups
liv_se$Treatment<- relevel(liv_se$Treatment, 
                           "Vehicle") # tells the system which group is "control"
liv_se <- liv_se[ rowSums(assay(liv_se)) >= 10, ] #remove genes that have a total count less than 10, a good prefilter measure
```

#DeSeq2
We can now construct a DESeqDataSet object to formulate a starting point for our analysis. You need to add an appropriate design for analysis
##liver
```{r}
dds <- DESeqDataSet(liv_se, 
                    design = ~ Treatment) 
```

The variance stabilizing transformation (VST) a goal of stablizing the variance across the range of values. produce log2-like values for high counts. 
```{r}
L_vsd <- vst(dds)
```

Now for visuals, ploat a principal components analysis (PCA) using  ggplot2
###PCA
```{r}
L_data <- plotPCA(L_vsd, 
                  intgroup ="Treatment", 
                  returnData=TRUE)

L_percentVar <- round(100 * attr(L_data, 
                                 "percentVar"))
```

```{r}
tiff(file = "PCA_plots/Liver_PCA.tiff", units="in", width=8, height=5, res=1000)
PCA <-ggplot(L_data, aes(PC1, 
                   PC2, 
                   color=Treatment,
                   shape= Treatment)) + 
  stat_ellipse(aes(color= Treatment))+
  ggtitle("Liver PCA")+
  geom_point(size=3) +
  scale_color_manual(values = c("#999900", "#FF9933", "#3399FF", "#9933FF")) +
  scale_shape_manual(values = c(19, 15, 17, 18)) +
  xlab(paste0("PC1: ",
              L_percentVar[1],
              "% variance")) +
  ylab(paste0("PC2: ",
              L_percentVar[2],
              "% variance"))+ 
  theme_bw() + 
  theme(panel.border = element_rect(fill=NULL, colour = "black",size=1),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(), 
        axis.line = element_line(colour = "black"))
PCA
dev.off()
PCA

```


###Differential Expression
Lets run a differential expression pipeline using the raw counts

This function will print out a message for the various steps it performs. These are described in more detail in the manual page for DESeq, which can be accessed by typing ?DESeq. Briefly these are: the estimation of size factors (controlling for differences in the sequencing depth of the samples), the estimation of dispersion values for each gene, and fitting a generalized linear model.
```{r}
ldds <- DESeq(dds)
lres1<-results(ldds, name="Treatment_6_vs_Vehicle")
lres2<-results(ldds, name="Treatment_1_vs_Vehicle")
lres3<-results(ldds, name="Treatment_0.1_vs_Vehicle")
```
Calling results without any arguments will extract the estimated log2 fold changes and p values for the last variable in the design formula. If there are more than 2 levels for this variable, results will extract the results table for a comparison of the last level over the first level. Treatment 6 vs vehicle
```{r}
summary(lres1)
summary(lres2)
summary(lres3)
```
#liver up and down DEGs

```{r}
sum(lres3$log2FoldChange >1 & lres3$padj< 0.1, na.rm=TRUE)

sum(lres3$log2FoldChange < -1 & lres3$padj< 0.1, na.rm=TRUE)
```

###annotating results

How to assign actual gene names to our counts. Using an annotation package for Mus musculus. 
```{r}
lres1$symbol <- mapIds(Mus.musculus, #which annotation package model to use
                     keys=row.names(lres1), #using rownames
                     column="SYMBOL", #add a column of symbols associated with row.names arguments for the model
                     keytype="GENEID", #our rownames are gene id's 
                     multiVals="first")
lres2$symbol <- mapIds(Mus.musculus, #which annotation package model to use
                     keys=row.names(lres2), #using rownames
                     column="SYMBOL", #add a column of symbols associated with row.names arguments for the model
                     keytype="GENEID", #our rownames are gene id's 
                     multiVals="first")
lres3$symbol <- mapIds(Mus.musculus, #which annotation package model to use
                     keys=row.names(lres3), #using rownames
                     column="SYMBOL", #add a column of symbols associated with row.names arguments for the model
                     keytype="GENEID", #our rownames are gene id's 
                     multiVals="first")

```

```{r}
lres1$TXname <- mapIds(Mus.musculus, #which annotation package model to use
                     keys=row.names(lres1), #using rownames
                     column="TXNAME", #add a column of symbols associated with row.names arguments for the model
                     keytype="GENEID", #our rownames are gene id's 
                     multiVals="first")
lres2$TXname <- mapIds(Mus.musculus, #which annotation package model to use
                     keys=row.names(lres2), #using rownames
                     column="TXNAME", #add a column of symbols associated with row.names arguments for the model
                     keytype="GENEID", #our rownames are gene id's 
                     multiVals="first")
lres3$TXname <- mapIds(Mus.musculus, #which annotation package model to use
                     keys=row.names(lres3), #using rownames
                     column="TXNAME", #add a column of symbols associated with row.names arguments for the model
                     keytype="GENEID", #our rownames are gene id's 
                     multiVals="first")

```

```{r}
lres1$entrez <- mapIds(Mus.musculus,
                     keys=row.names(lres1),
                     column="ENTREZID",
                     keytype="GENEID",
                     multiVals="first")
lres2$entrez <- mapIds(Mus.musculus,
                     keys=row.names(lres2),
                     column="ENTREZID",
                     keytype="GENEID",
                     multiVals="first")
lres3$entrez <- mapIds(Mus.musculus,
                     keys=row.names(lres3),
                     column="ENTREZID",
                     keytype="GENEID",
                     multiVals="first")
```

```{r}
lres1$Genename <- mapIds(Mus.musculus,
                     keys=row.names(lres1),
                     column="GENENAME", #now add a column for gene names or gene description
                     keytype="GENEID",
                     multiVals="first")
lres2$Genename <- mapIds(Mus.musculus,
                     keys=row.names(lres2),
                     column="GENENAME", #now add a column for gene names or gene description
                     keytype="GENEID",
                     multiVals="first")
lres3$Genename <- mapIds(Mus.musculus,
                     keys=row.names(lres3),
                     column="GENENAME", #now add a column for gene names or gene description
                     keytype="GENEID",
                     multiVals="first")
```

```{r}
lres1$GOID <- mapIds(Mus.musculus,
                     keys=row.names(lres1),
                     column="GO", #now add a column for gene names or gene description
                     keytype="GENEID",
                     multiVals="first")
lres2$GOID <- mapIds(Mus.musculus,
                     keys=row.names(lres2),
                     column="GO", #now add a column for gene names or gene description
                     keytype="GENEID",
                     multiVals="first")
lres3$GOID <- mapIds(Mus.musculus,
                     keys=row.names(lres3),
                     column="GO", #now add a column for gene names or gene description
                     keytype="GENEID",
                     multiVals="first")
```
```{r}
lres1$ensembl <- mapIds(Mus.musculus,
                     keys=row.names(lres1),
                     column="ENSEMBL", #now add a column for gene names or gene description
                     keytype="GENEID",
                     multiVals="first")
lres2$ensembl <- mapIds(Mus.musculus,
                     keys=row.names(lres2),
                     column="ENSEMBL", #now add a column for gene names or gene description
                     keytype="GENEID",
                     multiVals="first")
lres3$ensembl <- mapIds(Mus.musculus,
                     keys=row.names(lres3),
                     column="ENSEMBL", #now add a column for gene names or gene description
                     keytype="GENEID",
                     multiVals="first")
```

###exporting results
```{r}
write.csv( as.data.frame(lres1), file="Results_spreadsheets/L_veh_6_results.csv" )
write.csv( as.data.frame(lres2), file="Results_spreadsheets/L_veh_1_results.csv" )
write.csv( as.data.frame(lres3), file="Results_spreadsheets/L_veh_0.1_results.csv" )
```

```{r}
write_tsv( as.data.frame(lres1), file="Results_spreadsheets/L_veh_6_results.tsv" )
write_tsv( as.data.frame(lres2), file="Results_spreadsheets/L_veh_1_results.tsv" )
write_tsv( as.data.frame(lres3), file="Results_spreadsheets/L_veh_0.1_results.tsv" )
```

```{r}
assay(liv_se)
write.csv(as.data.frame(assay(liv_se)), file= "Results_spreadsheets/Liv_6mg_counts.csv")
```


```{r}
L_resOrdered1 <- lres1[order(lres1$padj),] #reorder the genes based on significance
L_resOrdered2 <- lres2[order(lres2$padj),] #reorder the genes based on significance
L_resOrdered3 <- lres3[order(lres3$padj),] #reorder the genes based on significance
```

###Plotting results
####Heatmap
Heatmaps are a good way to visualize the most significant genes
```{r}
L_mat <- assay(L_vsd)[ head(order(lres1$padj),
                            30), #top 30 genes 
                       ]
#L_mat <- L_mat - rowMeans(L_mat)
L_df <- as.data.frame(colData(L_vsd)[,"Treatment"])
colnames(L_df)[1]<- "Treatment"

```


```{r}
tiff(file = "Heatmaps/Liver_heatmap.tiff", units="in", width=8, height=5, res=1000)
heat<- pheatmap(L_mat, 
         annotation_col=L_df,
         labels_row = L_resOrdered1$symbol,
         fontsize = 6,
         scale = "row",
         show_colnames = F,
         main = "Liver Heatmap")

heat 

dev.off()

heat
```

Do the same analysis for the following tissues.

#Striatum Tissue
```{r}
Sinfo$samples<- gsub(".bam", "", Sinfo$ï..Sample_ID)
rownames(Sinfo)<-Sinfo$samples
colData(s_se)
colData(s_se)<-DataFrame(Sinfo)
s_se$Treatment<- as.factor(s_se$Treatment)
s_se$Treatment<- relevel(s_se$Treatment, 
                           "Vehicle")
s_se <- s_se[ rowSums(assay(s_se)) >= 10, ]
s_se$Mouse.ID
```

##DEG analysis
```{r}
sdds <- DESeqDataSet(s_se, 
                    design = ~ Treatment)
```

```{r}
s_vsd <- vst(sdds)
```

```{r}
s_data <- plotPCA(s_vsd, 
                  intgroup ="Treatment", 
                  returnData=TRUE)
s_data
s_percentVar <- round(100 * attr(s_data, 
                                 "percentVar"))
```

##PCA
```{r}
tiff(file = "PCA_plots/Striatum_PCA.tiff", units="in", width=8, height=5, res=1000)
PCA <-ggplot(s_data, aes(PC1, 
                   PC2, 
                   color=Treatment,
                   shape= Treatment)) + 
  stat_ellipse(aes(color= Treatment))+
  ggtitle("Striatum PCA")+
  geom_point(size=3) +
  scale_color_manual(values = c("#999900", "#FF9933", "#3399FF", "#9933FF")) +
  scale_shape_manual(values = c(19, 15, 17, 18)) +
  xlab(paste0("PC1: ",
              L_percentVar[1],
              "% variance")) +
  ylab(paste0("PC2: ",
              L_percentVar[2],
              "% variance"))+ 
  theme_bw() + 
  theme(panel.border = element_rect(fill=NULL, colour = "black",size=1),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(), 
        axis.line = element_line(colour = "black"))
PCA
dev.off()
PCA

```

```{r}
sdds <- DESeq(sdds)

sres1<-results(sdds, name="Treatment_6_vs_Vehicle")
sres2<-results(sdds, name="Treatment_1_vs_Vehicle")
sres3<-results(sdds, name="Treatment_0.1_vs_Vehicle")
```
Calling results without any arguments will extract the estimated log2 fold changes and p values for the last variable in the design formula. If there are more than 2 levels for this variable, results will extract the results table for a comparison of the last level over the first level. Treatment 6 vs vehicle
```{r}
summary(sres1)
summary(sres2)
summary(sres3)
sum(M_res$log2FoldChange > 1 & M_res$pvalue < 0.05, na.rm=TRUE)
```


```{r}
sum(sres3$log2FoldChange >0.3 & sres3$padj< 0.1, na.rm=TRUE)

sum(sres3$log2FoldChange < -0.3 & sres3$padj< 0.1, na.rm=TRUE)
```

##Annotating results
How to assign actual gene names to our counts. Using an annotation package for Mus musculus. 
```{r}
sres1$symbol <- mapIds(Mus.musculus, #which annotation package model to use
                     keys=row.names(sres1), #using rownames
                     column="SYMBOL", #add a column of symbols associated with row.names arguments for the model
                     keytype="GENEID", #our rownames are gene id's 
                     multiVals="first")
sres2$symbol <- mapIds(Mus.musculus, #which annotation package model to use
                     keys=row.names(sres2), #using rownames
                     column="SYMBOL", #add a column of symbols associated with row.names arguments for the model
                     keytype="GENEID", #our rownames are gene id's 
                     multiVals="first")
sres3$symbol <- mapIds(Mus.musculus, #which annotation package model to use
                     keys=row.names(sres3), #using rownames
                     column="SYMBOL", #add a column of symbols associated with row.names arguments for the model
                     keytype="GENEID", #our rownames are gene id's 
                     multiVals="first")

```

```{r}
sres1$TXname <- mapIds(Mus.musculus, #which annotation package model to use
                     keys=row.names(sres1), #using rownames
                     column="TXNAME", #add a column of symbols associated with row.names arguments for the model
                     keytype="GENEID", #our rownames are gene id's 
                     multiVals="first")
sres2$TXname <- mapIds(Mus.musculus, #which annotation package model to use
                     keys=row.names(sres2), #using rownames
                     column="TXNAME", #add a column of symbols associated with row.names arguments for the model
                     keytype="GENEID", #our rownames are gene id's 
                     multiVals="first")
sres3$TXname <- mapIds(Mus.musculus, #which annotation package model to use
                     keys=row.names(sres3), #using rownames
                     column="TXNAME", #add a column of symbols associated with row.names arguments for the model
                     keytype="GENEID", #our rownames are gene id's 
                     multiVals="first")

```

```{r}
sres1$entrez <- mapIds(Mus.musculus,
                     keys=row.names(sres1),
                     column="ENTREZID",
                     keytype="GENEID",
                     multiVals="first")
sres2$entrez <- mapIds(Mus.musculus,
                     keys=row.names(sres2),
                     column="ENTREZID",
                     keytype="GENEID",
                     multiVals="first")
sres3$entrez <- mapIds(Mus.musculus,
                     keys=row.names(sres3),
                     column="ENTREZID",
                     keytype="GENEID",
                     multiVals="first")
```

```{r}
sres1$Genename <- mapIds(Mus.musculus,
                     keys=row.names(sres1),
                     column="GENENAME", #now add a column for gene names or gene description
                     keytype="GENEID",
                     multiVals="first")
sres2$Genename <- mapIds(Mus.musculus,
                     keys=row.names(sres2),
                     column="GENENAME", #now add a column for gene names or gene description
                     keytype="GENEID",
                     multiVals="first")
sres3$Genename <- mapIds(Mus.musculus,
                     keys=row.names(sres3),
                     column="GENENAME", #now add a column for gene names or gene description
                     keytype="GENEID",
                     multiVals="first")
```

```{r}
sres1$GOID <- mapIds(Mus.musculus,
                     keys=row.names(sres1),
                     column="GO", #now add a column for gene names or gene description
                     keytype="GENEID",
                     multiVals="first")
sres2$GOID <- mapIds(Mus.musculus,
                     keys=row.names(sres2),
                     column="GO", #now add a column for gene names or gene description
                     keytype="GENEID",
                     multiVals="first")
sres3$GOID <- mapIds(Mus.musculus,
                     keys=row.names(sres3),
                     column="GO", #now add a column for gene names or gene description
                     keytype="GENEID",
                     multiVals="first")
```

```{r}
sres1$ensembl <- mapIds(Mus.musculus,
                     keys=row.names(sres1),
                     column="ENSEMBL", #now add a column for gene names or gene description
                     keytype="GENEID",
                     multiVals="first")
sres2$ensembl <- mapIds(Mus.musculus,
                     keys=row.names(sres2),
                     column="ENSEMBL", #now add a column for gene names or gene description
                     keytype="GENEID",
                     multiVals="first")
sres3$ensembl <- mapIds(Mus.musculus,
                     keys=row.names(sres3),
                     column="ENSEMBL", #now add a column for gene names or gene description
                     keytype="GENEID",
                     multiVals="first")
```
##exporting results
```{r}
write.csv( as.data.frame(sres1), file="Results_spreadsheets/S_veh_6_results.csv" )
write.csv( as.data.frame(sres2), file="Results_spreadsheets/S_veh_1_results.csv" )
write.csv( as.data.frame(sres3), file="Results_spreadsheets/S_veh_0.1_results.csv" )
```

```{r}
write_tsv( as.data.frame(sres1), file="Results_spreadsheets/S_veh_6_results.tsv" )
write_tsv( as.data.frame(sres2), file="Results_spreadsheets/S_veh_1_results.tsv" )
write_tsv( as.data.frame(sres3), file="Results_spreadsheets/S_veh_0.1_results.tsv" )
```


```{r}
write.csv(as.data.frame(assay(s_se)), file= "Results_spreadsheets/STN_6mg_counts.csv")
```

```{r}
S_resOrdered1 <- sres1[order(sres1$padj),] #reorder the genes based on significance
S_resOrdered2 <- sres2[order(sres2$padj),] #reorder the genes based on significance
S_resOrdered3 <- sres3[order(sres3$padj),] #reorder the genes based on significance
```

#visualization
```{r}
s_mat <- assay(s_vsd)[ head(order(sres1$padj),
                            30), ]
#s_mat <- s_mat - rowMeans(s_mat)
s_df <- as.data.frame(colData(s_vsd)[,
                                     c(
                                       "Treatment")])
colnames(s_df)[1]<- "Treatment"
```

```{r}
tiff(file = "Heatmaps/Striatum_heatmap.tiff", units="in", width=8, height=5, res=1000)
pheatmap(s_mat, 
         annotation_col=s_df,
         labels_row = S_resOrdered1$symbol,
         fontsize = 6,
         scale = "row",
         show_colnames = F,
         main = "Striatum Heatmap")
dev.off()
```

#***Piriform Tissue***

```{r}
pinfo$samples<- gsub(".bam", "", pinfo$ï..Sample_ID)
rownames(pinfo)<-pinfo$samples
colData(p_se)
colData(p_se)<-DataFrame(pinfo)
p_se$Treatment<- as.factor(p_se$Treatment)
p_se$Treatment<- relevel(p_se$Treatment, 
                           "Vehicle")
p_se <- p_se[ rowSums(assay(p_se)) >= 10, ]
```

```{r}
pdds <- DESeqDataSet(p_se, 
                    design = ~ Treatment)
```

```{r}
p_vsd <- vst(pdds)
```

```{r}
p_data <- plotPCA(p_vsd, 
                  intgroup ="Treatment", 
                  returnData=TRUE)
p_percentVar <- round(100 * attr(p_data, 
                                 "percentVar"))
```

#Prefrontal PCA
```{r}
tiff(file = "PCA_plots/PFC_PCA.tiff", units="in", width=8, height=5, res=1000)
PCA <-ggplot(p_data, aes(PC1, 
                   PC2, 
                   color=Treatment,
                   shape= Treatment)) + 
  stat_ellipse(aes(color= Treatment))+
  ggtitle("Prefrontal Cortex PCA")+
  geom_point(size=3) +
  scale_color_manual(values = c("#999900", "#FF9933", "#3399FF", "#9933FF")) +
  scale_shape_manual(values = c(19, 15, 17, 18)) +
  xlab(paste0("PC1: ",
              L_percentVar[1],
              "% variance")) +
  ylab(paste0("PC2: ",
              L_percentVar[2],
              "% variance"))+ 
  theme_bw() + 
  theme(panel.border = element_rect(fill=NULL, colour = "black",size=1),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(), 
        axis.line = element_line(colour = "black"))
PCA
dev.off()
PCA

```

```{r}
pdds <- DESeq(pdds)
pres1<-results(pdds, name="Treatment_6_vs_Vehicle")
pres2<-results(pdds, name="Treatment_1_vs_Vehicle")
pres3<-results(pdds, name="Treatment_0.1_vs_Vehicle")
```
Calling results without any arguments will extract the estimated log2 fold changes and p values for the last variable in the design formula. If there are more than 2 levels for this variable, results will extract the results table for a comparison of the last level over the first level. Treatment 6 vs vehicle
```{r}
summary(pres1)
summary(pres2)
summary(pres3)

```
```{r}
sum(pres1$log2FoldChange >0.3 & pres1$padj< 0.1, na.rm=TRUE)

sum(pres1$log2FoldChange < -0.3 & pres1$padj< 0.1, na.rm=TRUE)
```

#***Annotating results***
How to assign actual gene names to our counts. Using an annotation package for Mus musculus. 
```{r}
pres1$symbol <- mapIds(Mus.musculus, #which annotation package model to use
                     keys=row.names(pres1), #using rownames
                     column="SYMBOL", #add a column of symbols associated with row.names arguments for the model
                     keytype="GENEID", #our rownames are gene id's 
                     multiVals="first")
pres2$symbol <- mapIds(Mus.musculus, #which annotation package model to use
                     keys=row.names(pres2), #using rownames
                     column="SYMBOL", #add a column of symbols associated with row.names arguments for the model
                     keytype="GENEID", #our rownames are gene id's 
                     multiVals="first")
pres3$symbol <- mapIds(Mus.musculus, #which annotation package model to use
                     keys=row.names(pres3), #using rownames
                     column="SYMBOL", #add a column of symbols associated with row.names arguments for the model
                     keytype="GENEID", #our rownames are gene id's 
                     multiVals="first")

```

```{r}
pres1$TXname <- mapIds(Mus.musculus, #which annotation package model to use
                     keys=row.names(pres1), #using rownames
                     column="TXNAME", #add a column of symbols associated with row.names arguments for the model
                     keytype="GENEID", #our rownames are gene id's 
                     multiVals="first")
pres2$TXname <- mapIds(Mus.musculus, #which annotation package model to use
                     keys=row.names(pres2), #using rownames
                     column="TXNAME", #add a column of symbols associated with row.names arguments for the model
                     keytype="GENEID", #our rownames are gene id's 
                     multiVals="first")
pres3$TXname <- mapIds(Mus.musculus, #which annotation package model to use
                     keys=row.names(pres3), #using rownames
                     column="TXNAME", #add a column of symbols associated with row.names arguments for the model
                     keytype="GENEID", #our rownames are gene id's 
                     multiVals="first")

```

```{r}
pres1$entrez <- mapIds(Mus.musculus,
                     keys=row.names(pres1),
                     column="ENTREZID",
                     keytype="GENEID",
                     multiVals="first")
pres2$entrez <- mapIds(Mus.musculus,
                     keys=row.names(pres2),
                     column="ENTREZID",
                     keytype="GENEID",
                     multiVals="first")
pres3$entrez <- mapIds(Mus.musculus,
                     keys=row.names(pres3),
                     column="ENTREZID",
                     keytype="GENEID",
                     multiVals="first")
```

```{r}
pres1$Genename <- mapIds(Mus.musculus,
                     keys=row.names(pres1),
                     column="GENENAME", #now add a column for gene names or gene description
                     keytype="GENEID",
                     multiVals="first")
pres2$Genename <- mapIds(Mus.musculus,
                     keys=row.names(pres2),
                     column="GENENAME", #now add a column for gene names or gene description
                     keytype="GENEID",
                     multiVals="first")
pres3$Genename <- mapIds(Mus.musculus,
                     keys=row.names(pres3),
                     column="GENENAME", #now add a column for gene names or gene description
                     keytype="GENEID",
                     multiVals="first")
```

```{r}
pres1$GOID <- mapIds(Mus.musculus,
                     keys=row.names(pres1),
                     column="GO", #now add a column for gene names or gene description
                     keytype="GENEID",
                     multiVals="first")
pres2$GOID <- mapIds(Mus.musculus,
                     keys=row.names(pres2),
                     column="GO", #now add a column for gene names or gene description
                     keytype="GENEID",
                     multiVals="first")
pres3$GOID <- mapIds(Mus.musculus,
                     keys=row.names(pres3),
                     column="GO", #now add a column for gene names or gene description
                     keytype="GENEID",
                     multiVals="first")
```
```{r}
pres1$ensembl <- mapIds(Mus.musculus,
                     keys=row.names(pres1),
                     column="ENSEMBL", #now add a column for gene names or gene description
                     keytype="GENEID",
                     multiVals="first")
pres2$ensembl <- mapIds(Mus.musculus,
                     keys=row.names(pres2),
                     column="ENSEMBL", #now add a column for gene names or gene description
                     keytype="GENEID",
                     multiVals="first")
pres3$ensembl <- mapIds(Mus.musculus,
                     keys=row.names(pres3),
                     column="ENSEMBL", #now add a column for gene names or gene description
                     keytype="GENEID",
                     multiVals="first")
```
#exporting Prefrontal Cortex Results
```{r}
write.csv( as.data.frame(pres1), file="Results_spreadsheets/P_veh_6_results.csv" )
write.csv( as.data.frame(pres2), file="Results_spreadsheets/P_veh_1_results.csv" )
write.csv( as.data.frame(pres3), file="Results_spreadsheets/P_veh_0.1_results.csv" )
```

```{r}
write_tsv( as.data.frame(pres1), file="Results_spreadsheets/P_veh_6_results.tsv" )
write_tsv( as.data.frame(pres2), file="Results_spreadsheets/P_veh_1_results.tsv" )
write_tsv( as.data.frame(pres3), file="Results_spreadsheets/P_veh_0.1_results.tsv" )
```


```{r}
write.csv(as.data.frame(assay(p_se)), file= "Results_spreadsheets/PFC_6mg_counts.csv")
```


```{r}
P_resOrdered1 <- pres1[order(pres1$padj),] #reorder the genes based on significance
P_resOrdered2 <- pres2[order(pres2$padj),] #reorder the genes based on significance
P_resOrdered3 <- pres3[order(pres3$padj),] #reorder the genes based on significance
```

#visualization

```{r}
p_mat <- assay(p_vsd)[ head(order(pres1$padj),
                            30), ]
#p_mat <- p_mat - rowMeans(p_mat)
p_df <- as.data.frame(colData(p_vsd)[,
                                     c(
                                       "Treatment")])
colnames(p_df)[1]<- "Treatment"
```

```{r}
tiff(file = "Heatmaps/Prefrontal_cortex_heatmap.tiff", units="in", width=8, height=5, res=1000)
pheatmap(p_mat, 
         annotation_col=p_df,
         labels_row = P_resOrdered1$symbol,
         scale = "row",
         fontsize = 8,
         show_colnames = F,
         main = "Prefrontal Cortex Heatmap")
dev.off()
```

#***volcano plots***
```{r}
tiff(file = 'volcano_plots/L_volcano_plot_6vsVeh.tiff', units="in", width=8, height=5, res=1000)
L1<- EnhancedVolcano(lres1,
    lab = lres1$symbol,
    x = 'log2FoldChange',
    y = 'padj',
    title = "Liver",
    subtitle = "6 mg/kg",
    xlim = c(-7,7) ,
    ylim = c(0,18),
    FCcutoff = 1.0,
    pCutoff = 0.1,
    labSize = 3.0,
    colAlpha = 1,
    legendLabels=c('Not sig.','Log (base 2) FC','adj p-value',
      'adj p-value & Log (base 2) FC'),
    legendPosition = 'right',
    legendLabSize = 12,
    legendIconSize = 3.0,
    gridlines.major = FALSE,
    gridlines.minor = FALSE,
    border = 'full',
    borderWidth = 0.5,
    borderColour = 'black')
L1
dev.off()
L1
```
```{r}
tiff(file = 'volcano_plots/L_volcano_plot_1vsVeh.tiff', units="in", width=8, height=5, res=1000)
L2<- EnhancedVolcano(lres2,
    lab = lres2$symbol,
    x = 'log2FoldChange',
    y = 'padj',
    title = "Liver",
    subtitle = "1 mg/kg",
    xlim = c(-7,7) ,
    ylim = c(0,18),
    FCcutoff = 1.0,
    pCutoff = 0.1,
    labSize = 3.0,
    colAlpha = 1,
    legendLabels=c('Not sig.','Log (base 2) FC','adj p-value',
      'adj p-value & Log (base 2) FC'),
    legendPosition = 'right',
    legendLabSize = 12,
    legendIconSize = 3.0,
    gridlines.major = FALSE,
    gridlines.minor = FALSE,
    border = 'full',
    borderWidth = 0.5,
    borderColour = 'black')
L2
dev.off()
L2
```
```{r}
tiff(file = 'volcano_plots/L_volcano_plot_0.1vsVeh.tiff', units="in", width=8, height=5, res=1000)
L3<- EnhancedVolcano(lres3,
    lab = lres3$symbol,
    x = 'log2FoldChange',
    y = 'padj',
    title = "Liver",
    subtitle = "0.1 mg/kg",
    xlim = c(-7,7) ,
    ylim = c(0,18),
    FCcutoff = 1.0,
    pCutoff = 0.1,
    labSize = 3.0,
    colAlpha = 1,
    legendLabels=c('Not sig.','Log (base 2) FC','adj p-value',
      'adj p-value & Log (base 2) FC'),
    legendPosition = 'right',
    legendLabSize = 12,
    legendIconSize = 3.0,
    gridlines.major = FALSE,
    gridlines.minor = FALSE,
    border = 'full',
    borderWidth = 0.5,
    borderColour = 'black')
L3
dev.off()
L3
```


```{r}
tiff(file = 'volcano_plots/P_volcano_plot_6vsVeh.tiff', units="in", width=8, height=5, res=1000)
p1<- EnhancedVolcano(pres1,
                     lab =pres1$symbol,
                     x = 'log2FoldChange',
                     y = 'padj',
                     title = "Prefrontal Cortex",
                    subtitle = "6 mg/kg",
                     xlim = c(-2.5,2.5) ,
                     ylim = c(0,4),
                     FCcutoff = 0.3,
                     pCutoff = 0.1,
                     labSize = 3.0,
                     colAlpha = 1,
                     legendLabels=c('Not sig.',
                                    'Log (base 2) FC',
                                    'adj p-value',
                                    'adj p-value & Log (base 2) FC'),
                     legendPosition = 'right',
                     legendLabSize = 12,
                     legendIconSize = 3.0,
                     gridlines.major = FALSE,
                     gridlines.minor = FALSE,
                     border = 'full',
                     borderWidth = 0.5,
                     borderColour = 'black')
p1
dev.off()
p1
```

```{r}
tiff(file = 'volcano_plots/P_volcano_plot_1vsVeh.tiff', units="in", width=8, height=5, res=1000)
p2<- EnhancedVolcano(pres2,
                     lab = pres2$symbol,
                     x = 'log2FoldChange',
                      y = 'padj',
                     title = "Prefrontal Cortex",
                    subtitle = "1 mg/kg",
                     xlim = c(-2.5,2.5) ,
                     ylim = c(0,4),
                     FCcutoff = 0.3,
                     pCutoff = 0.1,
                     labSize = 3.0,
                     colAlpha = 1,
                     legendLabels=c('Not sig.',
                                    'Log (base 2) FC',
                                    'adj p-value',
                                    'adj p-value & Log (base 2) FC'),
                     legendPosition = 'right',
                     legendLabSize = 12,
                     legendIconSize = 3.0,
                     gridlines.major = FALSE,
                     gridlines.minor = FALSE,
                     border = 'full',
                     borderWidth = 0.5,
                     borderColour = 'black')
p2
dev.off()
p2
```
```{r}
tiff(file = 'volcano_plots/P_volcano_plot_0.1vsVeh.tiff', units="in", width=8, height=5, res=1000)
p3<- EnhancedVolcano(pres3,
                     lab = pres3$symbol,
                     x = 'log2FoldChange',
                      y = 'padj',
                     title = "Prefrontal Cortex",
                    subtitle = "0.1 mg/kg",
                     xlim = c(-2.5,2.5) ,
                     ylim = c(0,4),
                     FCcutoff = 0.3,
                     pCutoff = 0.1,
                     labSize = 3.0,
                     colAlpha = 1,
                     legendLabels=c('Not sig.',
                                    'Log (base 2) FC',
                                    'adj p-value',
                                    'adj p-value & Log (base 2) FC'),
                     legendPosition = 'right',
                     legendLabSize = 12,
                     legendIconSize = 3.0,
                     gridlines.major = FALSE,
                     gridlines.minor = FALSE,
                     border = 'full',
                     borderWidth = 0.5,
                     borderColour = 'black')
p3
dev.off()
p3
```


```{r}
tiff(file = 'volcano_plots/S_volcano_plot_6vsVeh.tiff', units="in", width=8, height=5, res=1000)
s1<-EnhancedVolcano(sres1,
    lab = sres1$symbol,
    x = 'log2FoldChange',
 y = 'padj',
                     title = "Striatum",
                    subtitle = "6 mg/kg",
                     xlim = c(-3,3) ,
                     ylim = c(0,2),
                     FCcutoff = 0.3,
                     pCutoff = 0.1,
                     labSize = 3.0,
                     colAlpha = 1,
                     legendLabels=c('Not sig.',
                                    'Log (base 2) FC',
                                    'adj p-value',
                                    'adj p-value & Log (base 2) FC'),
                     legendPosition = 'right',
    legendLabSize = 12,
    legendIconSize = 3.0,
    gridlines.major = FALSE,
    gridlines.minor = FALSE,
    border = 'full',
    borderWidth = 0.5,
    borderColour = 'black')
s1
dev.off()
s1
```
```{r}
tiff(file = 'volcano_plots/S_volcano_plot_1vsVeh.tiff', units="in", width=8, height=5, res=1000)
s2<-EnhancedVolcano(sres2,
    lab = sres2$symbol,
    x = 'log2FoldChange',
    y = 'padj',
                     title = "Striatum",
                    subtitle = "1 mg/kg",
                     xlim = c(-3,3) ,
                     ylim = c(0,2),
                     FCcutoff = 0.3,
                     pCutoff = 0.1,
                     labSize = 3.0,
                     colAlpha = 1,
                     legendLabels=c('Not sig.',
                                    'Log (base 2) FC',
                                    'adj p-value',
                                    'adj p-value & Log (base 2) FC'),
                     legendPosition = 'right',
    legendLabSize = 12,
    legendIconSize = 3.0,
    gridlines.major = FALSE,
    gridlines.minor = FALSE,
    border = 'full',
    borderWidth = 0.5,
    borderColour = 'black')
s2
dev.off()
s2
```
```{r}
tiff(file = 'volcano_plots/S_volcano_plot_0.1vsVeh.tiff', units="in", width=8, height=5, res=1000)
s3<-EnhancedVolcano(sres3,
    lab = sres3$symbol,
    x = 'log2FoldChange',
y = 'padj',
                     title = "Striatum",
                    subtitle = "0.1 mg/kg",
                     xlim = c(-3,3) ,
                     ylim = c(0,2),
                     FCcutoff = 0.3,
                     pCutoff = 0.1,
                     labSize = 3.0,
                     colAlpha = 1,
                     legendLabels=c('Not sig.',
                                    'Log (base 2) FC',
                                    'adj p-value',
                                    'adj p-value & Log (base 2) FC'),
                     legendPosition = 'right',
    legendLabSize = 12,
    legendIconSize = 3.0,
    gridlines.major = FALSE,
    gridlines.minor = FALSE,
    border = 'full',
    borderWidth = 0.5,
    borderColour = 'black')
s3
dev.off()
s3
```


```{r}
inputList <- list(rankedGenes1 = genes, rankedGenes2 = genes2)
test.out <- compareCluster(geneClusters=inputList,  fun = "gseKEGG", pvalueCutoff = 0.05, pAdjustMethod = "none")
```

```{r}
dotplot(test.out, showCategory=5, split=".sign") + facet_grid(.~.sign)
```


#gene enrichment
```{r}
x<- lres1[order(lres1$log2FoldChange, decreasing = TRUE),] #reorder the genes based on significance
gene_list<-x$log2FoldChange
names(gene_list)<-x$ensembl
head(gene_list)
```

```{r}
lgse <- gseGO(geneList=gene_list, 
             ont ="ALL", 
             keyType = "ENSEMBL", 
             nPerm = 10000, 
             minGSSize = 3, 
             maxGSSize = 800, 
             pvalueCutoff = 0.05, 
             verbose = TRUE, 
             OrgDb = org.Mm.eg.db, 
             pAdjustMethod = "none")

```
```{r}
tiff(file = 'gene_enrichment/Liver_6_enrichment.tiff', units="in", width=7, height=5, res=1000)
dot1<-dotplot(lgse, showCategory=5, split=".sign") +
  geom_point(shape = 1,colour = "black") +
  scale_color_viridis(option = "inferno", direction = -1) +
  facet_grid(.~.sign)+
  theme(panel.spacing = unit(0.5, "cm",data = NULL), panel.grid.minor = element_blank())+
  scale_y_discrete(labels = function(x) ifelse(str_detect(x, "gap"), "", str_wrap(x, width = 30)))+
  theme(axis.text.y = element_text(size = 14), 
        strip.background = element_blank(), strip.text.x = element_text(size = 16)) +
  theme(legend.text = element_text(size = 10),legend.key.width= unit(0.3, 'cm'))
dot1
dev.off()
dot1
```

```{r}
data<-data.frame(lgse)
data
write.csv(data,file="gene_enrichment/Liver_6_gene_enrichment.csv")
```

```{r}
xx<- lres2[order(lres2$log2FoldChange, decreasing = TRUE),] #reorder the genes based on significance
gene_list2<-xx$log2FoldChange
names(gene_list2)<-xx$ensembl
head(gene_list2)
```

```{r}
lgse <- gseGO(geneList=gene_list, 
             ont ="ALL", 
             keyType = "ENSEMBL", 
             nPerm = 10000, 
             minGSSize = 3, 
             maxGSSize = 800, 
             pvalueCutoff = 0.05, 
             verbose = TRUE, 
             OrgDb = org.Mm.eg.db, 
             pAdjustMethod = "none")

```
```{r}
tiff(file = 'gene_enrichment/Liver_1_enrichment.tiff', units="in", width=7, height=5, res=1000)
dot2<-dotplot(lgse, showCategory=5, split=".sign") +
  geom_point(shape = 1,colour = "black") +
  scale_color_viridis(option = "inferno", direction = -1) +
  facet_grid(.~.sign)+
  theme(panel.spacing = unit(0.5, "cm",data = NULL), panel.grid.minor = element_blank())+
  scale_y_discrete(labels = function(x) ifelse(str_detect(x, "gap"), "", str_wrap(x, width = 30)))+
  theme(axis.text.y = element_text(size = 10), 
        strip.background = element_blank(), strip.text.x = element_text(size = 16)) +
  theme(legend.text = element_text(size = 10),legend.key.width= unit(0.3, 'cm'))
dot2
dev.off()
dot2
```

```{r}
data<-data.frame(lgse)
data
write.csv(data,file="gene_enrichment/Liver_1_gene_enrichment.csv")
```

```{r}
xxx<- lres3[order(lres3$log2FoldChange, decreasing = TRUE),] #reorder the genes based on significance
gene_list3<-xxx$log2FoldChange
names(gene_list3)<-xxx$ensembl
head(gene_list3)
```

```{r}
lgse <- gseGO(geneList=gene_list, 
             ont ="ALL", 
             keyType = "ENSEMBL", 
             nPerm = 10000, 
             minGSSize = 3, 
             maxGSSize = 800, 
             pvalueCutoff = 0.05, 
             verbose = TRUE, 
             OrgDb = org.Mm.eg.db, 
             pAdjustMethod = "none")

```

```{r}
tiff(file = 'gene_enrichment/Liver_0.1_enrichment.tiff', units="in", width=7, height=5, res=1000)
dot3<-dotplot(lgse, showCategory=5, split=".sign") +
  geom_point(shape = 1,colour = "black") +
  scale_color_viridis(option = "inferno", direction = -1) +
  facet_grid(.~.sign)+
  theme(panel.spacing = unit(0.5, "cm",data = NULL), panel.grid.minor = element_blank())+
  scale_y_discrete(labels = function(x) ifelse(str_detect(x, "gap"), "", str_wrap(x, width = 30)))+
  theme(axis.text.y = element_text(size = 12), 
        strip.background = element_blank(), strip.text.x = element_text(size = 16)) +
  theme(legend.text = element_text(size = 10),legend.key.width= unit(0.3, 'cm'))
dot3
dev.off()
dot3
```

```{r}
data<-data.frame(lgse)
data
write.csv(data,file="gene_enrichment/Liver_0.1_gene_enrichment.csv")
```

```{r}
x<- sres1[order(sres1$log2FoldChange, decreasing = TRUE),] #reorder the genes based on significance
gene_list<-x$log2FoldChange
names(gene_list)<-x$ensembl
head(gene_list)
```

```{r}
sgse <- gseGO(geneList=gene_list, 
             ont ="ALL", 
             keyType = "ENSEMBL", 
             nPerm = 10000, 
             minGSSize = 3, 
             maxGSSize = 800, 
             pvalueCutoff = 0.05, 
             verbose = TRUE, 
             OrgDb = org.Mm.eg.db, 
             pAdjustMethod = "none")
```

```{r}
tiff(file = 'gene_enrichment/Striatum_6_gene_enrichment.tiff', units="in", width=8, height=5, res=1000)
dot4<-dotplot(sgse, showCategory=5, split=".sign") +
  geom_point(shape = 1,colour = "black") +
  scale_color_viridis(option = "inferno", direction = -1) +
  facet_grid(.~.sign)+
  theme(panel.spacing = unit(0.5, "cm",data = NULL), panel.grid.minor = element_blank())+
  scale_y_discrete(labels = function(x) ifelse(str_detect(x, "gap"), "", str_wrap(x, width = 30)))+
  theme(axis.text.y = element_text(size = 14), 
        strip.background = element_blank(), strip.text.x = element_text(size = 16)) +
  theme(legend.text = element_text(size = 10),legend.key.width= unit(0.3, 'cm'))
dot4
dev.off()
dot4
```

```{r}
data<-data.frame(sgse)
data
write.csv(data,file="gene_enrichment/STN_6_gene_enrichment.csv")
```

```{r}
x<- sres2[order(sres2$log2FoldChange, decreasing = TRUE),] #reorder the genes based on significance
gene_list<-x$log2FoldChange
names(gene_list)<-x$ensembl
head(gene_list)
```

```{r}
sgse <- gseGO(geneList=gene_list, 
             ont ="ALL", 
             keyType = "ENSEMBL", 
             nPerm = 10000, 
             minGSSize = 3, 
             maxGSSize = 800, 
             pvalueCutoff = 0.05, 
             verbose = TRUE, 
             OrgDb = org.Mm.eg.db, 
             pAdjustMethod = "none")
```

```{r}
tiff(file = 'gene_enrichment/Striatum_1_gene_enrichment.tiff', units="in", width=8, height=5, res=1000)
dot5<-dotplot(sgse, showCategory=5, split=".sign") +
  geom_point(shape = 1,colour = "black") +
  scale_color_viridis(option = "inferno", direction = -1) +
  facet_grid(.~.sign)+
  theme(panel.spacing = unit(0.5, "cm",data = NULL), panel.grid.minor = element_blank())+
  scale_y_discrete(labels = function(x) ifelse(str_detect(x, "gap"), "", str_wrap(x, width = 30)))+
  theme(axis.text.y = element_text(size = 10), 
        strip.background = element_blank(), strip.text.x = element_text(size = 16)) +
  theme(legend.text = element_text(size = 10),legend.key.width= unit(0.3, 'cm'))
dot5
dev.off()
dot5
```

```{r}
data<-data.frame(sgse)
data
write.csv(data,file="gene_enrichment/STN_1_gene_enrichment.csv")
```

```{r}
x<- sres3[order(sres3$log2FoldChange, decreasing = TRUE),] #reorder the genes based on significance
gene_list<-x$log2FoldChange
names(gene_list)<-x$ensembl
head(gene_list)
```

```{r}
sgse <- gseGO(geneList=gene_list, 
             ont ="ALL", 
             keyType = "ENSEMBL", 
             nPerm = 10000, 
             minGSSize = 3, 
             maxGSSize = 800, 
             pvalueCutoff = 0.05, 
             verbose = TRUE, 
             OrgDb = org.Mm.eg.db, 
             pAdjustMethod = "none")
```
```{r}
tiff(file = 'gene_enrichment/Striatum_0.1_gene_enrichment.tiff', units="in", width=8, height=5, res=1000)
dot6<-dotplot(sgse, showCategory=5, split=".sign") +
  geom_point(shape = 1,colour = "black") +
  scale_color_viridis(option = "inferno", direction = -1) +
  facet_grid(.~.sign)+
  theme(panel.spacing = unit(0.5, "cm",data = NULL), panel.grid.minor = element_blank())+
  scale_y_discrete(labels = function(x) ifelse(str_detect(x, "gap"), "", str_wrap(x, width = 30)))+
  theme(axis.text.y = element_text(size = 14), 
        strip.background = element_blank(), strip.text.x = element_text(size = 16)) +
  theme(legend.text = element_text(size = 10),legend.key.width= unit(0.3, 'cm'))
dot6
dev.off()
dot6
```


```{r}
data<-data.frame(sgse)
data
write.csv(data,file="gene_enrichment/STN_0.1_gene_enrichment.csv")
```

```{r}
x<- pres1[order(pres1$log2FoldChange, decreasing = TRUE),] #reorder the genes based on significance
gene_list<-x$log2FoldChange
names(gene_list)<-x$ensembl
head(gene_list)
```

```{r}
pgse <- gseGO(geneList=gene_list, 
             ont ="ALL", 
             keyType = "ENSEMBL", 
             nPerm = 10000, 
             minGSSize = 3, 
             maxGSSize = 800, 
             pvalueCutoff = 0.05, 
             verbose = TRUE, 
             OrgDb = org.Mm.eg.db, 
             pAdjustMethod = "none")
```

```{r}
tiff(file = 'gene_enrichment/Prefrontal_6_gene_enrichment.tiff', units="in", width=8, height=5, res=1000)
dot7<-dotplot(pgse, showCategory=5, split=".sign") +
  geom_point(shape = 1,colour = "black") +
  scale_color_viridis(option = "inferno", direction = -1) +
  facet_grid(.~.sign)+
  theme(panel.spacing = unit(0.5, "cm",data = NULL), panel.grid.minor = element_blank())+
  scale_y_discrete(labels = function(x) ifelse(str_detect(x, "gap"), "", str_wrap(x, width = 30)))+
  theme(axis.text.y = element_text(size = 14), 
        strip.background = element_blank(), strip.text.x = element_text(size = 16)) +
  theme(legend.text = element_text(size = 10),legend.key.width= unit(0.3, 'cm'))
dot7
dev.off()
dot7
```

```{r}
data<-data.frame(pgse)
data
write.csv(data,file="gene_enrichment/PFC_6_gene_enrichment.csv")
```

```{r}
x<- pres2[order(pres2$log2FoldChange, decreasing = TRUE),] #reorder the genes based on significance
gene_list<-x$log2FoldChange
names(gene_list)<-x$ensembl
head(gene_list)
```

```{r}
pgse <- gseGO(geneList=gene_list, 
             ont ="ALL", 
             keyType = "ENSEMBL", 
             nPerm = 10000, 
             minGSSize = 3, 
             maxGSSize = 800, 
             pvalueCutoff = 0.05, 
             verbose = TRUE, 
             OrgDb = org.Mm.eg.db, 
             pAdjustMethod = "none")
```

```{r}
tiff(file = 'gene_enrichment/Prefrontal_1_gene_enrichment.tiff', units="in", width=8, height=5, res=1000)
dot8<-dotplot(pgse, showCategory=5, split=".sign") +
  geom_point(shape = 1,colour = "black") +
  scale_color_viridis(option = "inferno", direction = -1) +
  facet_grid(.~.sign)+
  theme(panel.spacing = unit(0.5, "cm",data = NULL), panel.grid.minor = element_blank())+
  scale_y_discrete(labels = function(x) ifelse(str_detect(x, "gap"), "", str_wrap(x, width = 30)))+
  theme(axis.text.y = element_text(size = 14), 
        strip.background = element_blank(), strip.text.x = element_text(size = 16)) +
  theme(legend.text = element_text(size = 10),legend.key.width= unit(0.3, 'cm'))
dot8
dev.off()
dot8
```


```{r}
data<-data.frame(pgse)
data
write.csv(data,file="gene_enrichment/PFC_1_gene_enrichment.csv")
```

```{r}
x<- pres3[order(pres3$log2FoldChange, decreasing = TRUE),] #reorder the genes based on significance
gene_list<-x$log2FoldChange
names(gene_list)<-x$ensembl
head(gene_list)
```

```{r}
pgse <- gseGO(geneList=gene_list, 
             ont ="ALL", 
             keyType = "ENSEMBL", 
             nPerm = 10000, 
             minGSSize = 3, 
             maxGSSize = 800, 
             pvalueCutoff = 0.05, 
             verbose = TRUE, 
             OrgDb = org.Mm.eg.db, 
             pAdjustMethod = "none")
```

```{r}
tiff(file = 'gene_enrichment/Prefrontal_0.1_gene_enrichment.tiff', units="in", width=8, height=5, res=1000)
dot9<-dotplot(pgse, showCategory=5, split=".sign") +
  geom_point(shape = 1,colour = "black") +
  scale_color_viridis(option = "inferno", direction = -1) +
  facet_grid(.~.sign)+
  theme(panel.spacing = unit(0.5, "cm",data = NULL), panel.grid.minor = element_blank())+
  scale_y_discrete(labels = function(x) ifelse(str_detect(x, "gap"), "", str_wrap(x, width = 30)))+
  theme(axis.text.y = element_text(size = 10), 
        strip.background = element_blank(), strip.text.x = element_text(size = 16)) +
  theme(legend.text = element_text(size = 10),legend.key.width= unit(0.3, 'cm'))
dot9
dev.off()
dot9
```

```{r}
data<-data.frame(pgse)
data
write.csv(data,file="gene_enrichment/PFC_0.1_gene_enrichment.csv")
```

#KEGG pathview
```{r}
x<- lres1[order(lres1$log2FoldChange, decreasing = TRUE),] #reorder the genes based on significance
gene_list<-x$log2FoldChange
names(gene_list)<-x$entrez
head(gene_list)
```


```{r}
kegg_organism = "mmu"
kk2 <- gseKEGG(geneList     = gene_list,
               organism     = kegg_organism,
               nPerm        = 10000,
               minGSSize    = 3,
               maxGSSize    = 800,
               pvalueCutoff = 0.05,
               pAdjustMethod = "none",
               keyType       = "ncbi-geneid")
```


```{r}
#Produce the native KEGG plot (PNG)
liver_mito <- pathview(gene.data=gene_list, pathway.id="04020", species = kegg_organism)

```



```{r}
tiff(file = 'KEGG_enrichment/Liver_6_KEGG_enrichment.tiff', units="in", width=8, height=5, res=1000)
dot10<-dotplot(kk2, showCategory=5, split=".sign") +
  geom_point(shape = 1,colour = "black") +
  scale_color_viridis(option = "inferno", direction = -1) +
  facet_grid(.~.sign)+
  theme(panel.spacing = unit(0.5, "cm",data = NULL), panel.grid.minor = element_blank())+
  scale_y_discrete(labels = function(x) ifelse(str_detect(x, "gap"), "", str_wrap(x, width = 30)))+
  theme(axis.text.y = element_text(size = 10), 
        strip.background = element_blank(), strip.text.x = element_text(size = 16)) +
  theme(legend.text = element_text(size = 10),legend.key.width= unit(0.3, 'cm'))
dot10
dev.off()
dot10
```

```{r}
data<-data.frame(kk2)
data
write.csv(data,file="KEGG_enrichment/Liv_6_KEGG_enrichment.csv")

```

```{r}
xx<- lres2[order(lres2$log2FoldChange, decreasing = TRUE),] #reorder the genes based on significance
gene_list2<-xx$log2FoldChange
names(gene_list2)<-xx$entrez
head(gene_list2)
```
```{r}
#Produce the native KEGG plot (PNG)
liver_mito <- pathview(gene.data=gene_list, pathway.id="04020", species = kegg_organism)

```

```{r}
kegg_organism = "mmu"
kk3 <- gseKEGG(geneList     = gene_list2,
               organism     = kegg_organism,
               nPerm        = 10000,
               minGSSize    = 3,
               maxGSSize    = 800,
               pvalueCutoff = 0.05,
               pAdjustMethod = "none",
               keyType       = "ncbi-geneid")
```

```{r}
tiff(file = 'KEGG_enrichment/Liver_1_KEGG_enrichment.tiff', units="in", width=8, height=5, res=1000)
dot11<-dotplot(kk2, showCategory=5, split=".sign") +
  geom_point(shape = 1,colour = "black") +
  scale_color_viridis(option = "inferno", direction = -1) +
  facet_grid(.~.sign)+
  theme(panel.spacing = unit(0.5, "cm",data = NULL), panel.grid.minor = element_blank())+
  scale_y_discrete(labels = function(x) ifelse(str_detect(x, "gap"), "", str_wrap(x, width = 30)))+
  theme(axis.text.y = element_text(size = 10), 
        strip.background = element_blank(), strip.text.x = element_text(size = 16)) +
  theme(legend.text = element_text(size = 10),legend.key.width= unit(0.3, 'cm'))
dot11
dev.off()
dot11
```

```{r}
data<-data.frame(kk2)
data
write.csv(data,file="KEGG_enrichment/Liv_1_KEGG_enrichment.csv")

```

```{r}
xxx<- lres3[order(lres3$log2FoldChange, decreasing = TRUE),] #reorder the genes based on significance
gene_list3<-xxx$log2FoldChange
names(gene_list3)<-xxx$entrez
head(gene_list3)
```


```{r}
kegg_organism = "mmu"
kk4 <- gseKEGG(geneList     = gene_list3,
               organism     = kegg_organism,
               nPerm        = 10000,
               minGSSize    = 3,
               maxGSSize    = 800,
               pvalueCutoff = 0.05,
               pAdjustMethod = "none",
               keyType       = "ncbi-geneid")
```

```{r}
tiff(file = 'KEGG_enrichment/Liver_0.1_KEGG_enrichment.tiff', units="in", width=8, height=5, res=1000)
dot12<-dotplot(kk2, showCategory=5, split=".sign") +
  geom_point(shape = 1,colour = "black") +
  scale_color_viridis(option = "inferno", direction = -1) +
  facet_grid(.~.sign)+
  theme(panel.spacing = unit(0.5, "cm",data = NULL), panel.grid.minor = element_blank())+
  scale_y_discrete(labels = function(x) ifelse(str_detect(x, "gap"), "", str_wrap(x, width = 30)))+
  theme(axis.text.y = element_text(size = 10), 
        strip.background = element_blank(), strip.text.x = element_text(size = 16)) +
  theme(legend.text = element_text(size = 10),legend.key.width= unit(0.3, 'cm'))
dot12
dev.off()
dot12
```


```{r}
data<-data.frame(kk2)
data
write.csv(data,file="KEGG_enrichment/Liv_0.1_KEGG_enrichment.csv")

```

```{r}
x<- pres1[order(pres1$log2FoldChange, decreasing = TRUE),] #reorder the genes based on significance
gene_list<-x$log2FoldChange
names(gene_list)<-x$entrez
head(gene_list)
```


```{r}
kegg_organism = "mmu"
kk2 <- gseKEGG(geneList     = gene_list,
               organism     = kegg_organism,
               nPerm        = 10000,
               minGSSize    = 3,
               maxGSSize    = 800,
               pvalueCutoff = 0.05,
               pAdjustMethod = "none",
               keyType       = "ncbi-geneid")
```

```{r}
tiff(file = 'KEGG_enrichment/PFC_6_KEGG_enrichment.tiff', units="in", width=8, height=5, res=1000)
dot13<-dotplot(kk2, showCategory=5, split=".sign") +
  geom_point(shape = 1,colour = "black") +
  scale_color_viridis(option = "inferno", direction = -1) +
  facet_grid(.~.sign)+
  theme(panel.spacing = unit(0.5, "cm",data = NULL), panel.grid.minor = element_blank())+
  scale_y_discrete(labels = function(x) ifelse(str_detect(x, "gap"), "", str_wrap(x, width = 30)))+
  theme(axis.text.y = element_text(size = 10), 
        strip.background = element_blank(), strip.text.x = element_text(size = 16)) +
  theme(legend.text = element_text(size = 10),legend.key.width= unit(0.3, 'cm'))
dot13
dev.off()
dot13
```


```{r}
data<-data.frame(kk2)
data
write.csv(data,file="KEGG_enrichment/PFC_6_KEGG_enrichment.csv")

```

```{r}
x<- pres2[order(pres2$log2FoldChange, decreasing = TRUE),] #reorder the genes based on significance
gene_list<-x$log2FoldChange
names(gene_list)<-x$entrez
head(gene_list)
```


```{r}
kegg_organism = "mmu"
kk2 <- gseKEGG(geneList     = gene_list,
               organism     = kegg_organism,
               nPerm        = 10000,
               minGSSize    = 3,
               maxGSSize    = 800,
               pvalueCutoff = 0.05,
               pAdjustMethod = "none",
               keyType       = "ncbi-geneid")
```

```{r}
tiff(file = 'KEGG_enrichment/PFC_1_KEGG_enrichment.tiff', units="in", width=8, height=5, res=1000)
dot14<-dotplot(kk2, showCategory=5, split=".sign") +
  geom_point(shape = 1,colour = "black") +
  scale_color_viridis(option = "inferno", direction = -1) +
  facet_grid(.~.sign)+
  theme(panel.spacing = unit(0.5, "cm",data = NULL), panel.grid.minor = element_blank())+
  scale_y_discrete(labels = function(x) ifelse(str_detect(x, "gap"), "", str_wrap(x, width = 30)))+
  theme(axis.text.y = element_text(size = 10), 
        strip.background = element_blank(), strip.text.x = element_text(size = 16)) +
  theme(legend.text = element_text(size = 10),legend.key.width= unit(0.3, 'cm'))
dot14
dev.off()
dot14
```

```{r}
data<-data.frame(kk2)
data
write.csv(data,file="KEGG_enrichment/PFC_1_KEGG_enrichment.csv")

```

```{r}
x<- pres3[order(pres3$log2FoldChange, decreasing = TRUE),] #reorder the genes based on significance
gene_list<-x$log2FoldChange
names(gene_list)<-x$entrez
head(gene_list)
```


```{r}
kegg_organism = "mmu"
kk2 <- gseKEGG(geneList     = gene_list,
               organism     = kegg_organism,
               nPerm        = 10000,
               minGSSize    = 3,
               maxGSSize    = 800,
               pvalueCutoff = 0.05,
               pAdjustMethod = "none",
               keyType       = "ncbi-geneid")
```

```{r}
tiff(file = 'KEGG_enrichment/PFC_0.1_KEGG_enrichment.tiff', units="in", width=8, height=5, res=1000)
dot15<-dotplot(kk2, showCategory=5, split=".sign") +
  geom_point(shape = 1,colour = "black") +
  scale_color_viridis(option = "inferno", direction = -1) +
  facet_grid(.~.sign)+
  theme(panel.spacing = unit(0.5, "cm",data = NULL), panel.grid.minor = element_blank())+
  scale_y_discrete(labels = function(x) ifelse(str_detect(x, "gap"), "", str_wrap(x, width = 30)))+
  theme(axis.text.y = element_text(size = 10), 
        strip.background = element_blank(), strip.text.x = element_text(size = 16)) +
  theme(legend.text = element_text(size = 10),legend.key.width= unit(0.3, 'cm'))
dot15
dev.off()
dot15
```


```{r}
data<-data.frame(kk2)
data
write.csv(data,file="KEGG_enrichment/PFC_0.1_KEGG_enrichment.csv")

```


```{r}
x<- sres1[order(sres1$log2FoldChange, decreasing = TRUE),] #reorder the genes based on significance
gene_list<-x$log2FoldChange
names(gene_list)<-x$entrez
head(gene_list)
```


```{r}
kegg_organism = "mmu"
kk2 <- gseKEGG(geneList     = gene_list,
               organism     = kegg_organism,
               nPerm        = 10000,
               minGSSize    = 3,
               maxGSSize    = 800,
               pvalueCutoff = 0.05,
               pAdjustMethod = "none",
               keyType       = "ncbi-geneid")
```

```{r}
tiff(file = 'KEGG_enrichment/Striatum_6_KEGG_enrichment.tiff', units="in", width=8, height=5, res=1000)
dot16<-dotplot(kk2, showCategory=5, split=".sign") +
  geom_point(shape = 1,colour = "black") +
  scale_color_viridis(option = "inferno", direction = -1) +
  facet_grid(.~.sign)+
  theme(panel.spacing = unit(0.5, "cm",data = NULL), panel.grid.minor = element_blank())+
  scale_y_discrete(labels = function(x) ifelse(str_detect(x, "gap"), "", str_wrap(x, width = 30)))+
  theme(axis.text.y = element_text(size = 10), 
        strip.background = element_blank(), strip.text.x = element_text(size = 16)) +
  theme(legend.text = element_text(size = 10),legend.key.width= unit(0.3, 'cm'))
dot16
dev.off()
dot16
```

```{r}
data<-data.frame(kk2)
data
write.csv(data,file="KEGG_enrichment/STN_6_KEGG_enrichment.csv")

```

```{r}
x<- sres2[order(sres2$log2FoldChange, decreasing = TRUE),] #reorder the genes based on significance
gene_list<-x$log2FoldChange
names(gene_list)<-x$entrez
head(gene_list)
```


```{r}
kegg_organism = "mmu"
kk2 <- gseKEGG(geneList     = gene_list,
               organism     = kegg_organism,
               nPerm        = 10000,
               minGSSize    = 3,
               maxGSSize    = 800,
               pvalueCutoff = 0.05,
               pAdjustMethod = "none",
               keyType       = "ncbi-geneid")
```

```{r}
tiff(file = 'KEGG_enrichment/Striatum_1_KEGG_enrichment.tiff', units="in", width=8, height=5, res=1000)
dot17<-dotplot(kk2, showCategory=5, split=".sign") +
  geom_point(shape = 1,colour = "black") +
  scale_color_viridis(option = "inferno", direction = -1) +
  facet_grid(.~.sign)+
  theme(panel.spacing = unit(0.5, "cm",data = NULL), panel.grid.minor = element_blank())+
  scale_y_discrete(labels = function(x) ifelse(str_detect(x, "gap"), "", str_wrap(x, width = 30)))+
  theme(axis.text.y = element_text(size = 10), 
        strip.background = element_blank(), strip.text.x = element_text(size = 16)) +
  theme(legend.text = element_text(size = 10),legend.key.width= unit(0.3, 'cm'))
dot17
dev.off()
dot17
```

```{r}
data<-data.frame(kk2)
data
write.csv(data,file="KEGG_enrichment/STN_1_KEGG_enrichment.csv")

```


```{r}
x<- sres3[order(sres3$log2FoldChange, decreasing = TRUE),] #reorder the genes based on significance
gene_list<-x$log2FoldChange
names(gene_list)<-x$entrez
head(gene_list)
```
```{r}
#Produce the native KEGG plot (PNG)
liver_mito <- pathview(gene.data=gene_list, pathway.id="04020", species = kegg_organism)

```

```{r}
kegg_organism = "mmu"
kk2 <- gseKEGG(geneList     = gene_list,
               organism     = kegg_organism,
               nPerm        = 10000,
               minGSSize    = 3,
               maxGSSize    = 800,
               pvalueCutoff = 0.05,
               pAdjustMethod = "none",
               keyType       = "ncbi-geneid")
```

```{r}
tiff(file = 'KEGG_enrichment/Striatum_0.1_KEGG_enrichment.tiff', units="in", width=8, height=5, res=1000)
dot18<-dotplot(kk2, showCategory=5, split=".sign") +
  geom_point(shape = 1,colour = "black") +
  scale_color_viridis(option = "inferno", direction = -1) +
  facet_grid(.~.sign)+
  theme(panel.spacing = unit(0.5, "cm",data = NULL), panel.grid.minor = element_blank())+
  scale_y_discrete(labels = function(x) ifelse(str_detect(x, "gap"), "", str_wrap(x, width = 30)))+
  theme(axis.text.y = element_text(size = 10), 
        strip.background = element_blank(), strip.text.x = element_text(size = 16)) +
  theme(legend.text = element_text(size = 10),legend.key.width= unit(0.3, 'cm'))
dot18
dev.off()
dot18
```

```{r}
data<-data.frame(kk2)
data
write.csv(data,file="KEGG_enrichment/STN_0.1_KEGG_enrichment.csv")

```

#disease ontology (DO) Liver
```{r}
x<- lres1[order(lres1$log2FoldChange, decreasing = TRUE),]   #reorder the genes based on significance

x<-na.omit(x)

gene_list<-x$log2FoldChange
g_list<-data.frame(gene_list)
colnames(g_list)<-"LFC"

g_list$entrez<-x$entrez


hum_orthos<- orthologs(genes = g_list$entrez, species = "mouse", human = FALSE)

DO_data<-merge(g_list, hum_orthos, by.x = "entrez", by.y= "entrez", all.x = TRUE)

#rename the listed Fold changes for the genes with the human equivalent for DO
genes1<-DO_data$LFC
names(genes1)<-DO_data$human_entrez
genes1<-genes1[order(genes1, decreasing=TRUE)]
head(genes1,10)
```


```{r}
DO <- gseDO(gene          = genes,
           minGSSize     = 120,
           pvalueCutoff  = 0.2,
           pAdjustMethod = "BH",
           verbose       = FALSE)
head(DO)
```


```{r}
data<-data.frame(DO)
write.csv(data ,file="Disease_enrichment/liv_6_disease_enrichment.csv")

```

```{r}
tiff(file = 'Disease_enrichment/Liv_6_disease_enrichment.tiff', units="in", width=7, height=5, res=1000)
dot19<-dotplot(DO,showCategory=5, split=".sign") +
  geom_point(shape = 1,colour = "black") +
  scale_color_viridis(option = "inferno", direction = -1) +
  facet_grid(.~.sign)+
  theme(panel.spacing = unit(0.5, "cm",data = NULL), panel.grid.minor = element_blank())+
  scale_y_discrete(labels = function(x) ifelse(str_detect(x, "gap"), "", str_wrap(x, width = 30)))+
  theme(axis.text.y = element_text(size = 14), 
        strip.background = element_blank(), strip.text.x = element_text(size = 16)) +
  theme(legend.text = element_text(size = 10),legend.key.width= unit(0.3, 'cm'))  
dot19
dev.off()
dot19
```

```{r}
x<- lres2[order(lres2$log2FoldChange, decreasing = TRUE),]   #reorder the genes based on significance

x<-na.omit(x)

gene_list<-x$log2FoldChange
g_list<-data.frame(gene_list)
colnames(g_list)<-"LFC"

g_list$entrez<-x$entrez


hum_orthos<- orthologs(genes = g_list$entrez, species = "mouse", human = FALSE)

DO_data<-merge(g_list, hum_orthos, by.x = "entrez", by.y= "entrez", all.x = TRUE)

#rename the listed Fold changes for the genes with the human equivalent for DO
genes2<-DO_data$LFC
names(genes2)<-DO_data$human_entrez
genes2<-genes2[order(genes2, decreasing=TRUE)]
head(genes2,10)
```


```{r}
DO <- gseDO(gene          = genes,
           minGSSize     = 120,
           pvalueCutoff  = 0.2,
           pAdjustMethod = "BH",
           verbose       = FALSE)
head(DO)
```


```{r}
data<-data.frame(DO)
write.csv(data ,file="Disease_enrichment/liv_1_disease_enrichment.csv")

```

```{r}
tiff(file = 'Disease_enrichment/Liv_1_disease_enrichment.tiff', units="in", width=7, height=5, res=1000)
dot20<-dotplot(DO,showCategory=5, split=".sign") +
  geom_point(shape = 1,colour = "black") +
  scale_color_viridis(option = "inferno", direction = -1) +
  facet_grid(.~.sign)+
  theme(panel.spacing = unit(0.5, "cm",data = NULL), panel.grid.minor = element_blank())+
  scale_y_discrete(labels = function(x) ifelse(str_detect(x, "gap"), "", str_wrap(x, width = 30)))+
  theme(axis.text.y = element_text(size = 14), 
        strip.background = element_blank(), strip.text.x = element_text(size = 16)) +
  theme(legend.text = element_text(size = 10),legend.key.width= unit(0.3, 'cm'))  
dot20
dev.off()
dot20
```

```{r}
x<- lres3[order(lres3$log2FoldChange, decreasing = TRUE),]   #reorder the genes based on significance

x<-na.omit(x)

gene_list<-x$log2FoldChange
g_list<-data.frame(gene_list)
colnames(g_list)<-"LFC"

g_list$entrez<-x$entrez


hum_orthos<- orthologs(genes = g_list$entrez, species = "mouse", human = FALSE)

DO_data<-merge(g_list, hum_orthos, by.x = "entrez", by.y= "entrez", all.x = TRUE)

#rename the listed Fold changes for the genes with the human equivalent for DO
genes3<-DO_data$LFC
names(genes3)<-DO_data$human_entrez
genes3<-genes3[order(genes3, decreasing=TRUE)]
head(genes3,10)
```


```{r}
DO <- gseDO(gene          = genes,
           minGSSize     = 120,
           pvalueCutoff  = 0.2,
           pAdjustMethod = "BH",
           verbose       = FALSE)
head(DO)
```


```{r}
data<-data.frame(DO)
write.csv(data ,file="Disease_enrichment/liv_0.1_disease_enrichment.csv")

```

```{r}
tiff(file = 'Disease_enrichment/Liv_0.1_disease_enrichment.tiff', units="in", width=7, height=5, res=1000)
dot21<-dotplot(DO,showCategory=5, split=".sign") +
  geom_point(shape = 1,colour = "black") +
  scale_color_viridis(option = "inferno", direction = -1) +
  facet_grid(.~.sign)+
  theme(panel.spacing = unit(0.5, "cm",data = NULL), panel.grid.minor = element_blank())+
  scale_y_discrete(labels = function(x) ifelse(str_detect(x, "gap"), "", str_wrap(x, width = 30)))+
  theme(axis.text.y = element_text(size = 14), 
        strip.background = element_blank(), strip.text.x = element_text(size = 16)) +
  theme(legend.text = element_text(size = 10),legend.key.width= unit(0.3, 'cm'))  
dot21
dev.off()
dot21
```

#Compare Cluster
##*Liver*
###KEGG
```{r}
inputList <- list(`0.1` = genes3, `1` = genes2,`6` = genes1 )
Liv_groups <- compareCluster(geneClusters=inputList,  fun = "gseKEGG", pvalueCutoff = 0.05, pAdjustMethod = "none")
```


```{r}
tiff(file = 'KEGG_enrichment/Liver_grouped_kegg_enrichment.tiff', units="in", width=8, height=5, res=1000)
dot30<-dotplot(Liv_groups,showCategory=3, split=".sign") +
  geom_point(shape = 1,colour = "black") +
  scale_color_viridis(option = "inferno", direction = -1) +
  facet_grid(.~.sign)+
  theme(panel.spacing = unit(0.5, "cm",data = NULL), panel.grid.minor = element_blank())+
  scale_y_discrete(labels = function(x) ifelse(str_detect(x, "gap"), "", str_wrap(x, width = 45)))+
  theme(axis.text.y = element_text(size = 12), 
        strip.background = element_blank(),
        strip.text.x = element_text(size = 14),
        axis.text.x = element_text(size = 14),
        axis.title.x = element_blank(),
        legend.text = element_text(size = 10),
        legend.key.width= unit(0.3, 'cm'))  
dot30
dev.off()
dot30
```
###Disease
```{r}
Liv_groups_DO <- compareCluster(geneClusters=inputList,  fun = "gseDO", pvalueCutoff = 0.05, pAdjustMethod = "none")
```


```{r}
tiff(file = 'Disease_enrichment/Liver_grouped_disease_enrichment.tiff', units="in", width=8, height=5, res=1000)
dot33<-dotplot(Liv_groups_DO,showCategory=3, split=".sign") +
  geom_point(shape = 1,colour = "black") +
  scale_color_viridis(option = "inferno", direction = -1) +
  facet_grid(.~.sign)+
  theme(panel.spacing = unit(0.5, "cm",data = NULL), panel.grid.minor = element_blank())+
  scale_y_discrete(labels = function(x) ifelse(str_detect(x, "gap"), "", str_wrap(x, width = 45)))+
  theme(axis.text.y = element_text(size = 12), 
        strip.background = element_blank(),
        strip.text.x = element_text(size = 14),
        axis.text.x = element_text(size = 14),
        axis.title.x = element_blank(),
        legend.text = element_text(size = 10),
        legend.key.width= unit(0.3, 'cm'))  
dot33
dev.off()
dot33
```
###Gene
```{r}
liv_groups <- compareCluster(geneClusters=inputList,  fun = "gseGO", OrgDb=org.Hs.eg.db, pvalueCutoff = 0.05, pAdjustMethod = "none")
```


```{r}
tiff(file = 'gene_enrichment/liv_grouped_gene_enrichment.tiff', units="in", width=8, height=5, res=1000)
dot35<-dotplot(liv_groups,showCategory=3, split=".sign") +
  geom_point(shape = 1,colour = "black") +
  scale_color_viridis(option = "inferno", direction = -1) +
  facet_grid(.~.sign)+
  theme(panel.spacing = unit(0.5, "cm",data = NULL), panel.grid.minor = element_blank())+
  scale_y_discrete(labels = function(x) ifelse(str_detect(x, "gap"), "", str_wrap(x, width = 45)))+
  theme(axis.text.y = element_text(size = 10), 
        strip.background = element_blank(),
        strip.text.x = element_text(size = 14),
        axis.text.x = element_text(size = 14),
        axis.title.x = element_blank(),
        legend.text = element_text(size = 10),
        legend.key.width= unit(0.3, 'cm'))  
dot35
dev.off()
dot35
```

#Disease ontology
##PFC

```{r}
x<- pres1[order(pres1$log2FoldChange, decreasing = TRUE),]   #reorder the genes based on significance

x<-na.omit(x)

gene_list<-x$log2FoldChange
g_list<-data.frame(gene_list)
colnames(g_list)<-"LFC"

g_list$entrez<-x$entrez


hum_orthos<- orthologs(genes = g_list$entrez, species = "mouse", human = FALSE)

DO_data<-merge(g_list, hum_orthos, by.x = "entrez", by.y= "entrez", all.x = TRUE)

#rename the listed Fold changes for the genes with the human equivalent for DO
genes1<-DO_data$LFC
names(genes1)<-DO_data$human_entrez
genes1<-genes1[order(genes1, decreasing=TRUE)]
head(genes1,10)
```


```{r}
DO <- gseDO(gene          = genes,
           minGSSize     = 120,
           pvalueCutoff  = 0.2,
           pAdjustMethod = "BH",
           verbose       = FALSE)
head(DO)
```


```{r}
data<-data.frame(DO)
write.csv(data ,file="Disease_enrichment/PFC_6_disease_enrichment.csv")

```

```{r}
tiff(file = 'Disease_enrichment/PFC_6_disease_enrichment.tiff', units="in", width=7, height=5, res=1000)
dot22<-dotplot(DO,showCategory=5, split=".sign") +
  geom_point(shape = 1,colour = "black") +
  scale_color_viridis(option = "inferno", direction = -1) +
  facet_grid(.~.sign)+
  theme(panel.spacing = unit(0.5, "cm",data = NULL), panel.grid.minor = element_blank())+
  scale_y_discrete(labels = function(x) ifelse(str_detect(x, "gap"), "", str_wrap(x, width = 30)))+
  theme(axis.text.y = element_text(size = 14), 
        strip.background = element_blank(), strip.text.x = element_text(size = 16)) +
  theme(legend.text = element_text(size = 10),legend.key.width= unit(0.3, 'cm'))  
dot22
dev.off()
dot22
```

```{r}
x<- pres2[order(pres2$log2FoldChange, decreasing = TRUE),]   #reorder the genes based on significance

x<-na.omit(x)

gene_list<-x$log2FoldChange
g_list<-data.frame(gene_list)
colnames(g_list)<-"LFC"

g_list$entrez<-x$entrez


hum_orthos<- orthologs(genes = g_list$entrez, species = "mouse", human = FALSE)

DO_data<-merge(g_list, hum_orthos, by.x = "entrez", by.y= "entrez", all.x = TRUE)

#rename the listed Fold changes for the genes with the human equivalent for DO
genes2<-DO_data$LFC
names(genes2)<-DO_data$human_entrez
genes2<-genes2[order(genes2, decreasing=TRUE)]
head(genes2,10)
```


```{r}
DO <- gseDO(gene          = genes,
           minGSSize     = 120,
           pvalueCutoff  = 0.2,
           pAdjustMethod = "BH",
           verbose       = FALSE)
head(DO)
```


```{r}
data<-data.frame(DO)
write.csv(data ,file="Disease_enrichment/PFC_1_disease_enrichment.csv")

```

```{r}
tiff(file = 'Disease_enrichment/PFC_1_disease_enrichment.tiff', units="in", width=7, height=5, res=1000)
dot23<-dotplot(DO,showCategory=5, split=".sign") +
  geom_point(shape = 1,colour = "black") +
  scale_color_viridis(option = "inferno", direction = -1) +
  facet_grid(.~.sign)+
  theme(panel.spacing = unit(0.5, "cm",data = NULL), panel.grid.minor = element_blank())+
  scale_y_discrete(labels = function(x) ifelse(str_detect(x, "gap"), "", str_wrap(x, width = 30)))+
  theme(axis.text.y = element_text(size = 14), 
        strip.background = element_blank(), strip.text.x = element_text(size = 16)) +
  theme(legend.text = element_text(size = 10),legend.key.width= unit(0.3, 'cm'))  
dot23
dev.off()
dot23
```

```{r}
x<- pres3[order(pres3$log2FoldChange, decreasing = TRUE),]   #reorder the genes based on significance

x<-na.omit(x)

gene_list<-x$log2FoldChange
g_list<-data.frame(gene_list)
colnames(g_list)<-"LFC"

g_list$entrez<-x$entrez


hum_orthos<- orthologs(genes = g_list$entrez, species = "mouse", human = FALSE)

DO_data<-merge(g_list, hum_orthos, by.x = "entrez", by.y= "entrez", all.x = TRUE)

#rename the listed Fold changes for the genes with the human equivalent for DO
genes3<-DO_data$LFC
names(genes3)<-DO_data$human_entrez
genes3<-genes3[order(genes3, decreasing=TRUE)]
head(genes3,10)
```


```{r}
DO <- gseDO(gene          = genes,
           minGSSize     = 120,
           pvalueCutoff  = 0.2,
           pAdjustMethod = "BH",
           verbose       = FALSE)
head(DO)
```


```{r}
data<-data.frame(DO)
write.csv(data ,file="Disease_enrichment/PFC_0.1_disease_enrichment.csv")

```

```{r}
tiff(file = 'Disease_enrichment/PFC_0.1_disease_enrichment.tiff', units="in", width=7, height=5, res=1000)
dot24<-dotplot(DO,showCategory=5, split=".sign") +
  geom_point(shape = 1,colour = "black") +
  scale_color_viridis(option = "inferno", direction = -1) +
  facet_grid(.~.sign)+
  theme(panel.spacing = unit(0.5, "cm",data = NULL), panel.grid.minor = element_blank())+
  scale_y_discrete(labels = function(x) ifelse(str_detect(x, "gap"), "", str_wrap(x, width = 30)))+
  theme(axis.text.y = element_text(size = 14), 
        strip.background = element_blank(), strip.text.x = element_text(size = 16)) +
  theme(legend.text = element_text(size = 10),legend.key.width= unit(0.3, 'cm'))  
dot24
dev.off()
dot24
```

#Compare Cluster
##*PFC*
###Gene
```{r}
inputList <- list(`0.1` = genes3, `1` = genes2,`6` = genes1 )
PFC_groups <- compareCluster(geneClusters=inputList,  fun = "gseGO", OrgDb=org.Hs.eg.db, pvalueCutoff = 0.05, pAdjustMethod = "none")
```


```{r}
tiff(file = 'gene_enrichment/PFC_grouped_gene_enrichment.tiff', units="in", width=8, height=5, res=1000)
dot34<-dotplot(PFC_groups,showCategory=3, split=".sign") +
  geom_point(shape = 1,colour = "black") +
  scale_color_viridis(option = "inferno", direction = -1) +
  facet_grid(.~.sign)+
  theme(panel.spacing = unit(0.5, "cm",data = NULL), panel.grid.minor = element_blank())+
  scale_y_discrete(labels = function(x) ifelse(str_detect(x, "gap"), "", str_wrap(x, width = 45)))+
  theme(axis.text.y = element_text(size = 11), 
        strip.background = element_blank(),
        strip.text.x = element_text(size = 14),
        axis.text.x = element_text(size = 14),
        axis.title.x = element_blank(),
        legend.text = element_text(size = 10),
        legend.key.width= unit(0.3, 'cm'))  
dot34
dev.off()
dot34
```

###KEGG
```{r}
PFC_groups <- compareCluster(geneClusters=inputList,  fun = "gseKEGG", pvalueCutoff = 0.05, pAdjustMethod = "none")
```


```{r}
tiff(file = 'KEGG_enrichment/PFC_grouped_kegg_enrichment.tiff', units="in", width=8, height=5, res=1000)
dot28<-dotplot(PFC_groups,showCategory=3, split=".sign") +
  geom_point(shape = 1,colour = "black") +
  scale_color_viridis(option = "inferno", direction = -1) +
  facet_grid(.~.sign)+
  theme(panel.spacing = unit(0.5, "cm",data = NULL), panel.grid.minor = element_blank())+
  scale_y_discrete(labels = function(x) ifelse(str_detect(x, "gap"), "", str_wrap(x, width = 45)))+
  theme(axis.text.y = element_text(size = 14), 
        strip.background = element_blank(),
        strip.text.x = element_text(size = 14),
        axis.text.x = element_text(size = 14),
        axis.title.x = element_blank(),
        legend.text = element_text(size = 10),
        legend.key.width= unit(0.3, 'cm'))  
dot28
dev.off()
dot28
```

###Disease
```{r}
PFC_groups_DO <- compareCluster(geneClusters=inputList,  fun = "gseDO", pvalueCutoff = 0.05, pAdjustMethod = "none")
```


```{r}
tiff(file = 'Disease_enrichment/PFC_grouped_disease_enrichment.tiff', units="in", width=8, height=5, res=1000)
dot32<-dotplot(PFC_groups_DO,showCategory=3, split=".sign") +
  geom_point(shape = 1,colour = "black") +
  scale_color_viridis(option = "inferno", direction = -1) +
  facet_grid(.~.sign)+
  theme(panel.spacing = unit(0.5, "cm",data = NULL), panel.grid.minor = element_blank())+
  scale_y_discrete(labels = function(x) ifelse(str_detect(x, "gap"), "", str_wrap(x, width = 45)))+
  theme(axis.text.y = element_text(size = 12), 
        strip.background = element_blank(),
        strip.text.x = element_text(size = 14),
        axis.text.x = element_text(size = 14),
        axis.title.x = element_blank(),
        legend.text = element_text(size = 10),
        legend.key.width= unit(0.3, 'cm'))  
dot32
dev.off()
dot32
```


#larger Plots
##liver
```{r}
liver_big<- dot35/dot30/dot33

tiff(file = 'liver_dot_plots_big.tiff', units="in", width=10, height=15, res=500)
liver_big + plot_annotation(tag_levels = 'A') & 
  theme(plot.tag = element_text(size = 16))
dev.off()

```

##Striatum
```{r}
STN_big<- dot36/dot29/dot31


tiff(file = 'STN_dot_plots_big.tiff', units="in",width=10, height=15, res=500)
STN_big + plot_annotation(tag_levels = 'A') & 
  theme(plot.tag = element_text(size = 16))
dev.off()

```

##Prefrontal Cortex
```{r}
PFC_big<- dot34/dot28/dot32


tiff(file = 'PFC_dot_plots_big.tiff', units="in", width=10, height=15, res=500)
PFC_big + plot_annotation(tag_levels = 'A') & 
  theme(plot.tag = element_text(size = 16))
dev.off()

```


#Disease ontology STN
```{r}
x<- sres1[order(sres1$log2FoldChange, decreasing = TRUE),]   #reorder the genes based on significance

x<-na.omit(x)

gene_list<-x$log2FoldChange
g_list<-data.frame(gene_list)
colnames(g_list)<-"LFC"

g_list$entrez<-x$entrez


hum_orthos<- orthologs(genes = g_list$entrez, species = "mouse", human = FALSE)

DO_data<-merge(g_list, hum_orthos, by.x = "entrez", by.y= "entrez", all.x = TRUE)

#rename the listed Fold changes for the genes with the human equivalent for DO
genes1<-DO_data$LFC
names(genes1)<-DO_data$human_entrez
genes1<-genes1[order(genes1, decreasing=TRUE)]
head(genes1,10)
```


```{r}
DO <- gseDO(gene          = genes,
           minGSSize     = 120,
           pvalueCutoff  = 0.2,
           pAdjustMethod = "BH",
           verbose       = FALSE)
head(DO)
```


```{r}
data<-data.frame(DO)
write.csv(data ,file="Disease_enrichment/STN_6_disease_enrichment.csv")

```

```{r}
tiff(file = 'Disease_enrichment/STN_6_disease_enrichment.tiff', units="in", width=7, height=5, res=1000)
dot25<-dotplot(DO,showCategory=5, split=".sign") +
  geom_point(shape = 1,colour = "black") +
  scale_color_viridis(option = "inferno", direction = -1) +
  facet_grid(.~.sign)+
  theme(panel.spacing = unit(0.5, "cm",data = NULL), panel.grid.minor = element_blank())+
  scale_y_discrete(labels = function(x) ifelse(str_detect(x, "gap"), "", str_wrap(x, width = 30)))+
  theme(axis.text.y = element_text(size = 14), 
        strip.background = element_blank(), strip.text.x = element_text(size = 16)) +
  theme(legend.text = element_text(size = 10),legend.key.width= unit(0.3, 'cm'))  
dot25
dev.off()
dot25
```

```{r}
x<- sres2[order(sres2$log2FoldChange, decreasing = TRUE),]   #reorder the genes based on significance

x<-na.omit(x)

gene_list<-x$log2FoldChange
g_list<-data.frame(gene_list)
colnames(g_list)<-"LFC"

g_list$entrez<-x$entrez


hum_orthos<- orthologs(genes = g_list$entrez, species = "mouse", human = FALSE)

DO_data<-merge(g_list, hum_orthos, by.x = "entrez", by.y= "entrez", all.x = TRUE)

#rename the listed Fold changes for the genes with the human equivalent for DO
genes2<-DO_data$LFC
names(genes2)<-DO_data$human_entrez
genes2<-genes2[order(genes2, decreasing=TRUE)]
head(genes2,10)
```


```{r}
DO <- gseDO(gene          = genes,
           minGSSize     = 120,
           pvalueCutoff  = 0.2,
           pAdjustMethod = "BH",
           verbose       = FALSE)
head(DO)
```


```{r}
data<-data.frame(DO)
write.csv(data ,file="Disease_enrichment/STN_1_disease_enrichment.csv")

```

```{r}
tiff(file = 'Disease_enrichment/STN_1_disease_enrichment.tiff', units="in", width=7, height=5, res=1000)
dot26<-dotplot(DO,showCategory=5, split=".sign") +
  geom_point(shape = 1,colour = "black") +
  scale_color_viridis(option = "inferno", direction = -1) +
  facet_grid(.~.sign)+
  theme(panel.spacing = unit(0.5, "cm",data = NULL), panel.grid.minor = element_blank())+
  scale_y_discrete(labels = function(x) ifelse(str_detect(x, "gap"), "", str_wrap(x, width = 30)))+
  theme(axis.text.y = element_text(size = 14), 
        strip.background = element_blank(), strip.text.x = element_text(size = 16)) +
  theme(legend.text = element_text(size = 10),legend.key.width= unit(0.3, 'cm'))  
dot26
dev.off()
dot26
```

```{r}
x<- sres3[order(sres3$log2FoldChange, decreasing = TRUE),]   #reorder the genes based on significance

x<-na.omit(x)

gene_list<-x$log2FoldChange
g_list<-data.frame(gene_list)
colnames(g_list)<-"LFC"

g_list$entrez<-x$entrez


hum_orthos<- orthologs(genes = g_list$entrez, species = "mouse", human = FALSE)

DO_data<-merge(g_list, hum_orthos, by.x = "entrez", by.y= "entrez", all.x = TRUE)

#rename the listed Fold changes for the genes with the human equivalent for DO
genes3<-DO_data$LFC
names(genes3)<-DO_data$human_entrez
genes3<-genes3[order(genes3, decreasing=TRUE)]
head(genes3,10)
```


```{r}
DO <- gseDO(gene          = genes,
           minGSSize     = 120,
           pvalueCutoff  = 0.2,
           pAdjustMethod = "BH",
           verbose       = FALSE)
head(DO)
```


```{r}
data<-data.frame(DO)
write.csv(data ,file="Disease_enrichment/STN_0.1_disease_enrichment.csv")

```

```{r}
tiff(file = 'Disease_enrichment/STN_0.1_disease_enrichment.tiff', units="in", width=8, height=5, res=1000)
dot27<-dotplot(DO,showCategory=5, split=".sign") +
  geom_point(shape = 1,colour = "black") +
  scale_color_viridis(option = "inferno", direction = -1) +
  facet_grid(.~.sign)+
  theme(panel.spacing = unit(0.5, "cm",data = NULL), panel.grid.minor = element_blank())+
  scale_y_discrete(labels = function(x) ifelse(str_detect(x, "gap"), "", str_wrap(x, width = 45)))+
  theme(axis.text.y = element_text(size = 14), 
        strip.background = element_blank(), strip.text.x = element_text(size = 16)) +
  theme(legend.text = element_text(size = 10),legend.key.width= unit(0.3, 'cm'))  
dot27
dev.off()
dot27
```

#Compare Cluster
##*STN*
###Gene
```{r}
inputList <- list(`0.1` = genes3, `1` = genes2,`6` = genes1 )
STN_groups <- compareCluster(geneClusters=inputList,  fun = "gseGO", OrgDb=org.Hs.eg.db, pvalueCutoff = 0.05, pAdjustMethod = "none")
```


```{r}
tiff(file = 'gene_enrichment/STN_grouped_gene_enrichment.tiff', units="in", width=8, height=5, res=1000)
dot36<-dotplot(STN_groups,showCategory=3, split=".sign") +
  geom_point(shape = 1,colour = "black") +
  scale_color_viridis(option = "inferno", direction = -1) +
  facet_grid(.~.sign)+
  theme(panel.spacing = unit(0.5, "cm",data = NULL), panel.grid.minor = element_blank())+
  scale_y_discrete(labels = function(x) ifelse(str_detect(x, "gap"), "", str_wrap(x, width = 45)))+
  theme(axis.text.y = element_text(size = 11), 
        strip.background = element_blank(),
        strip.text.x = element_text(size = 14),
        axis.text.x = element_text(size = 14),
        axis.title.x = element_blank(),
        legend.text = element_text(size = 10),
        legend.key.width= unit(0.3, 'cm'))  
dot36
dev.off()
dot36
```

###KEGG
```{r}
STN_groups <- compareCluster(geneClusters=inputList,  fun = "gseKEGG", pvalueCutoff = 0.05, pAdjustMethod = "none")
```


```{r}
tiff(file = 'KEGG_enrichment/STN_grouped_kegg_enrichment.tiff', units="in", width=8, height=5, res=1000)
dot29<-dotplot(STN_groups,showCategory=3, split=".sign") +
  geom_point(shape = 1,colour = "black") +
  scale_color_viridis(option = "inferno", direction = -1) +
  facet_grid(.~.sign)+
  theme(panel.spacing = unit(0.5, "cm",data = NULL), panel.grid.minor = element_blank())+
  scale_y_discrete(labels = function(x) ifelse(str_detect(x, "gap"), "", str_wrap(x, width = 45)))+
  theme(axis.text.y = element_text(size = 13), 
        strip.background = element_blank(),
        strip.text.x = element_text(size = 14),
        axis.text.x = element_text(size = 14),
        axis.title.x = element_blank(),
        legend.text = element_text(size = 10),
        legend.key.width= unit(0.3, 'cm'))  
dot29
dev.off()
dot29
```


#Compare cluster STN DO
```{r}
STN_groups_DO <- compareCluster(geneClusters=inputList,  fun = "gseDO", pvalueCutoff = 0.05, pAdjustMethod = "none")
```


```{r}
tiff(file = 'Disease_enrichment/STN_grouped_disease_enrichment.tiff', units="in", width=8, height=5, res=1000)
dot31<-dotplot(STN_groups_DO,showCategory=3, split=".sign") +
  geom_point(shape = 1,colour = "black") +
  scale_color_viridis(option = "inferno", direction = -1) +
  facet_grid(.~.sign)+
  theme(panel.spacing = unit(0.5, "cm",data = NULL), panel.grid.minor = element_blank())+
  scale_y_discrete(labels = function(x) ifelse(str_detect(x, "gap"), "", str_wrap(x, width = 45)))+
  theme(axis.text.y = element_text(size = 12), 
        strip.background = element_blank(),
        strip.text.x = element_text(size = 14),
        axis.text.x = element_text(size = 14),
        axis.title.x = element_blank(),
        legend.text = element_text(size = 10),
        legend.key.width= unit(0.3, 'cm'))  
dot31
dev.off()
dot31
```




#plotting individual genes of interest (Liver)
Using normalized counts (prebuilt into DESEQ2) which normalizes counts by the estimated size factors (or normalization factors if these were used) and adds a pseudocount of 1/2 to allow for log scale plotting.
**drug metabolizing enzymes**
```{r}
top_gene<-read.csv("C://Users/agrebinoski/OneDrive - University of Iowa/MARBLES/gene_list_drug_metabolizing_kegg.csv", header = FALSE)
top_gene<-as.character(top_gene$V1)

setdiff(top_gene,rownames(counts(ldds))) #check if all genes of interest can be found in the rownames of the deseq object --> if not you will not be able to run the block of code. Character(0) or integer(0) is a what you want to see
```
```{r}
stat.test<-read.csv("Liv_FDR.csv")
stat.test$symbol<-factor(stat.test$symbol)
stat.test$group2<-as.factor(stat.test$group2)
stat.test$group1<-as.factor(stat.test$group1)
stat.test
```

```{r}
data_1 <- split(as.tibble(stat.test[,2:5]), as.factor(stat.test$symbol))
```


```{r message=FALSE}
plot_list<-list()
for (i in unique(1:length(top_gene))){
  gene <- top_gene[i]
  b<- plotCounts(ldds, gene = gene, intgroup = "Treatment",normalized = TRUE, returnData = TRUE)
  z = max(b$count) + (max(b$count)*0.05)
  d <- ggplot(b, aes(factor(Treatment), count))+
    geom_violin(mapping = aes(x = Treatment, 
                              y = count, 
                              fill = Treatment))+
    #adding jitter to avoid overplotting
    geom_point(mapping = aes(x = Treatment, 
                             y = count, 
                             fill = Treatment, 
                             shape= Treatment), 
               size = 5, position = position_jitter(width = 0.3, height=0))+ 
    scale_y_continuous(expand = c(0,0) , limits = c(0,z+z*0.4)) +
    stat_summary(mapping = aes(x = Treatment, y = count), 
                 geom = 'crossbar', 
                 fun= 'mean', 
                 colour = 'black', 
                 width=0.2)+
  scale_fill_manual(values = c("#999900", "#FF9933", "#3399FF", "#9933FF")) +
  scale_shape_manual(values = c(19, 15, 17, 18)) +
  labs(x = NULL, y = "Normalized counts")+ # changes the y axis label, removes x axis label
  theme_classic(base_size = 20)+# changes the overall style of the plot
  ggtitle(paste0(lres1$symbol[gene])) +
  theme(
    axis.text.x = element_text(colour = "black"),
    axis.text.y = element_text(colour = "black"))+
  stat_pvalue_manual(data_1[[i]], 
    y.position = z, step.increase = 0.2,
    label = "p = {scales::pvalue(p.adj)}", hide.ns = TRUE
    )
  plot_list[[gene]] <- d
}

head(plot_list)
```


```{r warning=FALSE}
# Export into pdf: display multiple plots on the same page
ggexport(
  plotlist = plot_list, filename = "gene_plots/Liver_cyp_plot.pdf", 
  ncol = 1, nrow = 1, height = 7, width = 7, res = 600,pointsize = 8
)
```


#Brain Regions
```{r}
top_gene<-read.csv("C://Users/agrebinoski/OneDrive - University of Iowa/MARBLES/gene_list_drug_metabolizing_kegg_striatum.csv", header = FALSE)
top_gene<-as.character(top_gene$V1)

setdiff(top_gene,rownames(counts(sdds))) #check if all genes of interest can be found in the rownames of the deseq object --> if not you will not be able to run the block of code. Character(0) or integer(0) is a what you want to see
```

```{r}
stat.test<-read.csv("STN_FDR.csv")
stat.test$symbol<-factor(stat.test$symbol)
stat.test$group2<-as.factor(stat.test$group2)
stat.test$group1<-as.factor(stat.test$group1)
stat.test
```

```{r}
data_1 <- split(as.tibble(stat.test[,2:5]), as.factor(stat.test$symbol))
```

```{r message=FALSE}
S_plot_list<-list()
for (i in unique(1:length(top_gene))){
  gene <- top_gene[i]
  b<- plotCounts(sdds, gene = gene, intgroup = "Treatment",normalized = TRUE, returnData = TRUE)
  z = max(b$count) + (max(b$count)*0.05)
  d <- ggplot(b, aes(factor(Treatment), count))+
    geom_violin(mapping = aes(x = Treatment, 
                              y = count, 
                              fill = Treatment))+
    #adding jitter to avoid overplotting
    geom_point(mapping = aes(x = Treatment, 
                             y = count, 
                             fill = Treatment, 
                             shape= Treatment), 
               size = 5, position = position_jitter(width = 0.3, height=0))+ 
    scale_y_continuous(expand = c(0,0) , limits = c(0,z+z*0.4)) +
    stat_summary(mapping = aes(x = Treatment, y = count), 
                 geom = 'crossbar', 
                 fun= 'mean', 
                 colour = 'black', 
                 width=0.2)+
  scale_fill_manual(values = c("#999900", "#FF9933", "#3399FF", "#9933FF")) +
  scale_shape_manual(values = c(19, 15, 17, 18)) +
  labs(x = NULL, y = "Normalized counts")+ # changes the y axis label, removes x axis label
  theme_classic(base_size = 20)+# changes the overall style of the plot
  ggtitle(paste0(sres1$symbol[gene])) +
  theme(
    axis.text.x = element_text(colour = "black"),
    axis.text.y = element_text(colour = "black"))+
  stat_pvalue_manual(data_1[[i]], 
    y.position = z, step.increase = 0.2,
    label = "p = {scales::pvalue(p.adj)}", hide.ns = TRUE
    )
  S_plot_list[[gene]] <- d
}

head(S_plot_list)
```

```{r warning=FALSE}
# Export into pdf: display multiple plots on the same page
ggexport(
  plotlist = S_plot_list, filename = "gene_plots/Striatum_cyp_plot.pdf", 
  ncol = 1, nrow = 1, height = 7, width = 7, res = 600,pointsize = 8
)
```

#prefrontal Cortex

```{r}
top_gene<-read.csv("C://Users/agrebinoski/OneDrive - University of Iowa/MARBLES/gene_list_drug_metabolizing_kegg_PFC.csv", header = FALSE)
top_gene<-as.character(top_gene$V1)

setdiff(top_gene,rownames(counts(pdds))) #check if all genes of interest can be found in the rownames of the deseq object --> if not you will not be able to run the block of code. Character(0) or integer(0) is a what you want to see
```

```{r}
stat.test<-read.csv("PFC_FDR.csv")
stat.test$symbol<-factor(stat.test$symbol)
stat.test$group2<-as.factor(stat.test$group2)
stat.test$group1<-as.factor(stat.test$group1)
stat.test
```

```{r}
data_1 <- split(as.tibble(stat.test[,2:5]), as.factor(stat.test$symbol))
```

```{r message=FALSE}
p_plot_list<-list()
for (i in unique(1:length(top_gene))){
  gene <- top_gene[i]
  b<- plotCounts(pdds, gene = gene, intgroup = "Treatment",normalized = TRUE, returnData = TRUE)
  z = max(b$count) + (max(b$count)*0.05)
  d <- ggplot(b, aes(factor(Treatment), count))+
    geom_violin(mapping = aes(x = Treatment, 
                              y = count, 
                              fill = Treatment))+
    #adding jitter to avoid overplotting
    geom_point(mapping = aes(x = Treatment, 
                             y = count, 
                             fill = Treatment, 
                             shape= Treatment), 
               size = 5, position = position_jitter(width = 0.3, height=0))+ 
    scale_y_continuous(expand = c(0,0) , limits = c(0,z+z*0.4)) +
    stat_summary(mapping = aes(x = Treatment, y = count), 
                 geom = 'crossbar', 
                 fun= 'mean', 
                 colour = 'black', 
                 width=0.2)+
  scale_fill_manual(values = c("#999900", "#FF9933", "#3399FF", "#9933FF")) +
  scale_shape_manual(values = c(19, 15, 17, 18)) +
  labs(x = NULL, y = "Normalized counts")+ # changes the y axis label, removes x axis label
  theme_classic(base_size = 20)+# changes the overall style of the plot
  ggtitle(paste0(pres1$symbol[gene])) +
  theme(
    axis.text.x = element_text(colour = "black"),
    axis.text.y = element_text(colour = "black"))+
  stat_pvalue_manual(data_1[[i]], 
    y.position = z, step.increase = 0.2,
    label = "p = {scales::pvalue(p.adj)}", hide.ns = TRUE
    )
  p_plot_list[[gene]] <- d
}

head(p_plot_list)
```

```{r warning=FALSE}
# Export into pdf: display multiple plots on the same page
ggexport(
  plotlist = p_plot_list, filename = "gene_plots/prefrontal_cyp_plot.pdf", 
  ncol = 1, nrow = 1, height = 7, width = 7, res = 600,pointsize = 8
)
```


#deconvolution - single cell

```{r}
sce.sce<- TasicBrainData(ensembl = TRUE) #single cell data set for the mouse brain
cellcluster<-factor(sce.sce$broad_type)
```


```{r}
p_se1<-p_se
key<-as.data.frame(pres1$ensembl)
colnames(key)[1]<-"ensembl"
key$unique<-make.names(key$ensembl, unique = TRUE)

bulk.mtx<-as(p_se1, "ExpressionSet") #my bulk RNA seq data for the mouse brain
bulk.mtx$Treatment
table(bulk.mtx$Treatment)
featureNames(bulk.mtx)<-key$unique
head(featureNames(bulk.mtx))


bulk.control.mtx = exprs(bulk.mtx)[, bulk.mtx$Treatment=='Vehicle']
bulk.case.mtx1 = exprs(bulk.mtx)[, bulk.mtx$Treatment=='6']
bulk.case.mtx2 = exprs(bulk.mtx)[, bulk.mtx$Treatment=='1']
bulk.case.mtx3 = exprs(bulk.mtx)[, bulk.mtx$Treatment=='0.1']

bulkcase<-cbind(bulk.case.mtx1,bulk.case.mtx2,bulk.case.mtx3)

```

```{r}
Est_all_groups= music2_prop_t_statistics(bulk.control.mtx = bulk.control.mtx, bulk.case.mtx = bulkcase, sc.sce = sce.sce, clusters = 'broad_type',samples = 'sample_title', select.ct =c("Astrocyte","Microglia","Oligodendrocyte", "Oligodendrocyte Precursor Cell","GABA-ergic Neuron", "Glutamatergic Neuron", "Endothelial Cell"), sample_prop=0.5,cutoff_c=0.05,cutoff_r=0.01)
```

```{r}
est.prop<- Est_all_groups$Est.prop
```

```{r}
prop_all = cbind('proportion'=c(est.prop), 'sample_title'=rep(rownames(est.prop),times=ncol(est.prop)), 'broad_type'=rep(colnames(est.prop), each=nrow(est.prop)))
prop_all
prop_all = as.data.frame(prop_all)
prop_all$proportion = as.numeric(as.character(prop_all$proportion))
prop_all$group = ifelse(prop_all$sample_title %in% seq(from=3, to=8, by=1),'6 mg/kg',
                        ifelse(prop_all$sample_title %in% seq(from=9, to=14, by=1),'1 mg/kg',
                        ifelse(prop_all$sample_title %in% seq(from=17, to=21, by=1),'0.1 mg/kg','Vehicle')))

prop_all$group<-factor(prop_all$group, levels = c("Vehicle","0.1 mg/kg", "1 mg/kg","6 mg/kg"))

#cols <-c("Astrocyte" = "cadetblue2", "Microglia" = "lightsalmon1", "GABA-ergic Neuron" = "palegreen2","Oligodendrocyte" = "goldenrod1", "Oligodendrocyte Precursor Cell"="steelblue3", "Glutamatergic Neuron" = "plum2","Endothelial Cell" = "tan" )

cols <-c("Vehicle" = "#999900", "0.1 mg/kg" = "#FF9933", "1 mg/kg" = "#3399FF","6 mg/kg" = "#9933FF")
```

```{r}
prop_all$proportion[81]<-NA
stat.test<-aov(proportion~group+broad_type+group*broad_type, prop_all)
tukey_hsd(cell_anova)

stat.test
prop_all$log_proportion<-log(prop_all$proportion)

cell_model<-lm(log_proportion~group+broad_type+group:broad_type, prop_all)
plot(cell_model, which=2)
cell_anova<-aov(cell_model)

sum<-tukey_hsd(cell_anova, which = "group:broad_type")
sum


```




```{r}
tiff(file = 'single_cell_plot/prefrontal_cortex_single_cell.tiff', units="in", width=10, height=8, res=1000)


scplot<- ggplot(prop_all, aes(x=group, y=log(proportion))) + xlab('')+
  ylab('Log of Cell Type Proportions')+
  theme_bw()+
  geom_violin(mapping = aes(x = group, y = log(proportion), fill = group))+
  geom_point(mapping = aes(x = group, y = log(proportion), shape= group), 
             size = 2, position = position_jitter(width = 0.3, height=0))+
  scale_shape_manual(values = c(19, 15, 17, 18))+
  stat_summary(fun = median,
               geom = "crossbar", width = 0.3,size=0.1,color='black')+
  theme(plot.title = element_text(hjust = 0.5, size=12),
        axis.text.x = element_blank(),
        axis.ticks.x = element_blank(),
        axis.text.y = element_text(size=12),
        axis.title.x = element_text(size=12),
        axis.title.y = element_text(size=12),
        axis.line = element_line(colour = "black"),
        strip.text.x = element_text(size = 12, face = "bold" ),
        strip.background = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = ,
        panel.background = element_blank(),
        legend.position = 'right')+
  scale_fill_manual(values = c("#999900", "#FF9933", "#3399FF", "#9933FF"))+ 
  facet_wrap(.~broad_type, labeller = label_wrap_gen(20), scales = "free")

scplot

dev.off()

scplot
```

```{r}
s_se1<-s_se
key<-as.data.frame(sres1$ensembl)
colnames(key)[1]<-"ensembl"
key$unique<-make.names(key$ensembl, unique = TRUE)

bulk.mtx<-as(s_se1, "ExpressionSet") #my bulk RNA seq data for the mouse brain
bulk.mtx$Treatment
table(bulk.mtx$Treatment)
featureNames(bulk.mtx)<-key$unique
head(featureNames(bulk.mtx))


bulk.control.mtx = exprs(bulk.mtx)[, bulk.mtx$Treatment=='Vehicle']
bulk.case.mtx1 = exprs(bulk.mtx)[, bulk.mtx$Treatment=='6']
bulk.case.mtx2 = exprs(bulk.mtx)[, bulk.mtx$Treatment=='1']
bulk.case.mtx3 = exprs(bulk.mtx)[, bulk.mtx$Treatment=='0.1']

bulkcase<-cbind(bulk.case.mtx1,bulk.case.mtx2,bulk.case.mtx3)

```

```{r}
Est_all_groups_stn= music2_prop_t_statistics(bulk.control.mtx = bulk.control.mtx, bulk.case.mtx = bulkcase, sc.sce = sce.sce, clusters = 'broad_type',samples = 'sample_title', select.ct =c("Astrocyte","Microglia","Oligodendrocyte", "Oligodendrocyte Precursor Cell","GABA-ergic Neuron", "Glutamatergic Neuron", "Endothelial Cell"), sample_prop=0.5,cutoff_c=0.05,cutoff_r=0.01)
```

```{r}
est.prop<- Est_all_groups_stn$Est.prop
```

```{r}
prop_all = cbind('proportion'=c(est.prop), 'sample_title'=rep(rownames(est.prop),times=ncol(est.prop)), 'broad_type'=rep(colnames(est.prop), each=nrow(est.prop)))
prop_all
prop_all = as.data.frame(prop_all)
prop_all$proportion = as.numeric(as.character(prop_all$proportion))
prop_all$group = ifelse(prop_all$sample_title %in% seq(from=3, to=8, by=1),'6 mg/kg',
                        ifelse(prop_all$sample_title %in% seq(from=9, to=14, by=1),'1 mg/kg',
                        ifelse(prop_all$sample_title %in% seq(from=17, to=21, by=1),'0.1 mg/kg','Vehicle')))
prop_all$group<-factor(prop_all$group, levels = c("Vehicle","0.1 mg/kg", "1 mg/kg","6 mg/kg"))
#cols <-c("Astrocyte" = "cadetblue2", "Microglia" = "lightsalmon1", "GABA-ergic Neuron" = "palegreen2","Oligodendrocyte" = "goldenrod1", "Oligodendrocyte Precursor Cell"="steelblue3", "Glutamatergic Neuron" = "plum2","Endothelial Cell" = "tan" )

cols <-c("Vehicle" = "#999900", "0.1 mg/kg" = "#FF9933", "1 mg/kg" = "#3399FF","6 mg/kg" = "#9933FF")
```

```{r}
prop_all$log_proportion<-log(prop_all$proportion)

cell_model<-lm(log_proportion~group+broad_type+group:broad_type, prop_all)
plot(cell_model, which=2)
cell_anova<-aov(cell_model)

tukey_hsd(cell_anova, which = "group:broad_type")

```


```{r}
tiff(file = 'single_cell_plot/Striatum_single_cell.tiff', units="in", width=10, height=8, res=1000)


scplot<- ggplot(prop_all, aes(x=group, y=log(proportion))) + xlab('')+
  ylab('Log of Cell Type Proportions')+
  theme_bw()+
  geom_violin(mapping = aes(x = group, y = log(proportion), fill = group))+
  geom_point(mapping = aes(x = group, y = log(proportion), shape= group), 
             size = 2, position = position_jitter(width = 0.3, height=0))+
  scale_shape_manual(values = c(19, 15, 17, 18))+
  stat_summary(fun = median,
               geom = "crossbar", width = 0.3,size=0.1,color='black')+
  theme(plot.title = element_text(hjust = 0.5, size=12),
        axis.text.x = element_blank(),
        axis.ticks.x = element_blank(),
        axis.text.y = element_text(size=12),
        axis.title.x = element_text(size=12),
        axis.title.y = element_text(size=12),
        axis.line = element_line(colour = "black"),
        strip.text.x = element_text(size = 12, face = "bold" ),
        strip.background = element_blank(),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = ,
        panel.background = element_blank(),
        legend.position = 'right')+
  scale_fill_manual(values = c("#999900", "#FF9933", "#3399FF", "#9933FF"))+ 
  facet_wrap(.~broad_type, labeller = label_wrap_gen(20), scales = "free")

scplot

dev.off()

scplot
```